// @saas-ui/theme/dist/anatomy/index.js
'use client'
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/anatomy/index.ts
var anatomy_exports = {};
__export(anatomy_exports, {
  appShellAnatomy: () => appShellAnatomy,
  bannerAnatomy: () => bannerAnatomy,
  emptyStateAnatomy: () => emptyStateAnatomy,
  hotkeysAnantomy: () => hotkeysAnantomy,
  loadingOverlayAnatomy: () => loadingOverlayAnatomy,
  navGroupAnatomy: () => navGroupAnatomy,
  navItemAnatomy: () => navItemAnatomy,
  nprogressAnatomy: () => nprogressAnatomy,
  personaAnatomy: () => personaAnatomy,
  propertyAnatomy: () => propertyAnatomy,
  searchInputAnatomy: () => searchInputAnatomy,
  selectAnatomy: () => selectAnatomy,
  sidebarAnatomy: () => sidebarAnatomy,
  stepperAnatomy: () => stepperAnatomy,
  structuredListAnatomy: () => structuredListAnatomy,
  timelineAnatomy: () => timelineAnatomy
});
module.exports = __toCommonJS(anatomy_exports);
var import_theme_tools = require("@chakra-ui/theme-tools");
var appShellAnatomy = (0, import_theme_tools.anatomy)("app-shell").parts(
  "container",
  "inner",
  "main"
);
var emptyStateAnatomy = (0, import_theme_tools.anatomy)("emptystate").parts(
  "container",
  "body",
  "icon",
  "title",
  "descripton",
  "actions",
  "footer"
);
var bannerAnatomy = (0, import_theme_tools.anatomy)("banner").parts(
  "container",
  "icon",
  "content",
  "title",
  "description",
  "actions",
  "close"
);
var hotkeysAnantomy = (0, import_theme_tools.anatomy)("hotkeys").parts(
  "container",
  "group",
  "groupTitle",
  "item",
  "command",
  "then"
);
var loadingOverlayAnatomy = (0, import_theme_tools.anatomy)("loading-overlay").parts(
  "overlay",
  "text"
);
var navGroupAnatomy = (0, import_theme_tools.anatomy)("nav-group").parts(
  "container",
  "title",
  "icon",
  "content"
);
var navItemAnatomy = (0, import_theme_tools.anatomy)("nav-item").parts(
  "item",
  "link",
  "inner",
  "icon",
  "label"
);
var nprogressAnatomy = (0, import_theme_tools.anatomy)("nprogress").parts("container", "bar");
var personaAnatomy = (0, import_theme_tools.anatomy)("persona").parts(
  "container",
  "details",
  "avatar",
  "label",
  "secondaryLabel",
  "tertiaryLabel"
);
var searchInputAnatomy = (0, import_theme_tools.anatomy)("search-input").parts(
  "input",
  "reset"
);
var sidebarAnatomy = (0, import_theme_tools.anatomy)("sidebar").parts(
  "container",
  "overlay",
  "section",
  "toggleWrapper",
  "toggle"
);
var stepperAnatomy = (0, import_theme_tools.anatomy)("stepper").parts(
  "container",
  "steps",
  "icon",
  "content",
  "title",
  "separator"
);
var structuredListAnatomy = (0, import_theme_tools.anatomy)("structured-list").parts(
  "list",
  "item",
  "button",
  "header",
  "cell",
  "icon"
);
var propertyAnatomy = (0, import_theme_tools.anatomy)("property").parts(
  "property",
  "label",
  "value"
);
var selectAnatomy = (0, import_theme_tools.anatomy)("select").parts(
  "addon",
  "field",
  "element"
);
var timelineAnatomy = (0, import_theme_tools.anatomy)("timeline").parts(
  "container",
  "item",
  "separator",
  "icon",
  "dot",
  "track",
  "content"
);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  appShellAnatomy,
  bannerAnatomy,
  emptyStateAnatomy,
  hotkeysAnantomy,
  loadingOverlayAnatomy,
  navGroupAnatomy,
  navItemAnatomy,
  nprogressAnatomy,
  personaAnatomy,
  propertyAnatomy,
  searchInputAnatomy,
  selectAnatomy,
  sidebarAnatomy,
  stepperAnatomy,
  structuredListAnatomy,
  timelineAnatomy
});
