// @zag-js/dom-utils/dist/index.mjs
import {
  getScrollOffset,
  getScrollParent,
  getScrollParents,
  isScrollParent
} from "./chunk-GXCY7Z36.mjs";
import {
  findByTypeahead
} from "./chunk-3RGJWUTP.mjs";
import {
  trackDocumentVisibility
} from "./chunk-LVLTWHTE.mjs";
import {
  trackVisualViewport
} from "./chunk-EUTSM3ED.mjs";
import {
  setVisuallyHidden,
  visuallyHiddenStyle
} from "./chunk-AVTS2722.mjs";
import {
  waitFor,
  waitForEvent
} from "./chunk-XM2BY4K7.mjs";
import {
  findByText,
  indexOfId,
  itemById,
  nextById,
  prevById,
  query,
  queryAll,
  sortByTreeOrder
} from "./chunk-G2LDCPYT.mjs";
import {
  trackPointerDown,
  trackPointerMove
} from "./chunk-5MDAOOLP.mjs";
import {
  disableTextSelection,
  restoreTextSelection
} from "./chunk-T5RLSDNR.mjs";
import {
  addPointerlockChangeListener,
  addPointerlockErrorListener,
  requestPointerLock
} from "./chunk-MC5E3AM7.mjs";
import {
  queueBeforeEvent
} from "./chunk-4LCPE4X5.mjs";
import {
  queueMicrotask
} from "./chunk-PBVZ5MDY.mjs";
import {
  observeElementRect
} from "./chunk-6CO5E7JW.mjs";
import {
  getEventPoint
} from "./chunk-NBQR4MDN.mjs";
import {
  getPointPercentRelativeToNode,
  getPointRelativeToNode,
  normalizePointValue
} from "./chunk-TTQLKCML.mjs";
import {
  getEventKey,
  getEventStep
} from "./chunk-SHYSWTB3.mjs";
import {
  addDomEvent,
  addPointerEvent,
  extractClientInfo,
  extractInfo,
  getEventName
} from "./chunk-XNMAYA7J.mjs";
import {
  observeAttributes,
  observeChildren
} from "./chunk-EKBMVORF.mjs";
import {
  disposable,
  disposableNextTick,
  disposableRaf,
  nextTick,
  raf
} from "./chunk-36SNO7OB.mjs";
import {
  ariaAttr,
  dataAttr,
  matchAttr
} from "./chunk-Q5SGM5OR.mjs";
import {
  MAX_Z_INDEX
} from "./chunk-UBLS5AGF.mjs";
import {
  copyVisualStyles
} from "./chunk-YJZ3V4BW.mjs";
import {
  getNativeEvent,
  isContextMenuEvent,
  isCtrlKey,
  isKeyboardClick,
  isLeftClick,
  isModifiedEvent,
  isMouseEvent,
  isPrintableKey,
  isRightClick,
  isSelfEvent,
  isTouchEvent,
  isVirtualClick,
  isVirtualPointerEvent,
  supportsMouseEvent,
  supportsPointerEvent,
  supportsTouchEvent
} from "./chunk-2HPIH2IE.mjs";
import {
  getPlatform,
  isApple,
  isDom,
  isFirefox,
  isIPhone,
  isIos,
  isMac,
  isSafari,
  isTouchDevice
} from "./chunk-V3PG4N2P.mjs";
import {
  fireBlurEvent,
  fireClickEvent,
  fireCustomEvent,
  fireKeyboardEvent
} from "./chunk-YGKMNMEU.mjs";
import {
  focusableSelector,
  getFirstFocusable,
  getFirstTabbable,
  getFocusables,
  getLastTabbable,
  getTabbables,
  isFocusable,
  isTabbable
} from "./chunk-GBGL7FGQ.mjs";
import {
  contains,
  defineDomHelpers,
  getActiveDescendant,
  getActiveElement,
  getDocument,
  getDocumentElement,
  getEventTarget,
  getEventWindow,
  getNodeName,
  getParent,
  getRootNode,
  getWindow,
  isDisabled,
  isDocument,
  isElementEditable,
  isFrame,
  isHTMLElement,
  isShadowRoot,
  isVisible,
  isWindow,
  isWithinShadowRoot
} from "./chunk-CFKU37OD.mjs";
import {
  getComputedStyle
} from "./chunk-VUVAE7SR.mjs";
import {
  getElementOffset
} from "./chunk-LYGZVQ73.mjs";
export {
  MAX_Z_INDEX,
  addDomEvent,
  addPointerEvent,
  addPointerlockChangeListener,
  addPointerlockErrorListener,
  ariaAttr,
  contains,
  copyVisualStyles,
  dataAttr,
  defineDomHelpers,
  disableTextSelection,
  disposable,
  disposableNextTick,
  disposableRaf,
  extractClientInfo,
  extractInfo,
  findByText,
  findByTypeahead,
  fireBlurEvent,
  fireClickEvent,
  fireCustomEvent,
  fireKeyboardEvent,
  focusableSelector,
  getActiveDescendant,
  getActiveElement,
  getComputedStyle,
  getDocument,
  getDocumentElement,
  getElementOffset,
  getEventKey,
  getEventName,
  getEventPoint,
  getEventStep,
  getEventTarget,
  getEventWindow,
  getFirstFocusable,
  getFirstTabbable,
  getFocusables,
  getLastTabbable,
  getNativeEvent,
  getNodeName,
  getParent,
  getPlatform,
  getPointPercentRelativeToNode,
  getPointRelativeToNode,
  getRootNode,
  getScrollOffset,
  getScrollParent,
  getScrollParents,
  getTabbables,
  getWindow,
  indexOfId,
  isApple,
  isContextMenuEvent,
  isCtrlKey,
  isDisabled,
  isDocument,
  isDom,
  isElementEditable,
  isFirefox,
  isFocusable,
  isFrame,
  isHTMLElement,
  isIPhone,
  isIos,
  isKeyboardClick,
  isLeftClick,
  isMac,
  isModifiedEvent,
  isMouseEvent,
  isPrintableKey,
  isRightClick,
  isSafari,
  isScrollParent,
  isSelfEvent,
  isShadowRoot,
  isTabbable,
  isTouchDevice,
  isTouchEvent,
  isVirtualClick,
  isVirtualPointerEvent,
  isVisible,
  isWindow,
  isWithinShadowRoot,
  itemById,
  matchAttr,
  nextById,
  nextTick,
  normalizePointValue,
  observeAttributes,
  observeChildren,
  observeElementRect,
  prevById,
  query,
  queryAll,
  queueBeforeEvent,
  queueMicrotask,
  raf,
  requestPointerLock,
  restoreTextSelection,
  setVisuallyHidden,
  sortByTreeOrder,
  supportsMouseEvent,
  supportsPointerEvent,
  supportsTouchEvent,
  trackDocumentVisibility,
  trackPointerDown,
  trackPointerMove,
  trackVisualViewport,
  visuallyHiddenStyle,
  waitFor,
  waitForEvent
};
