// @saas-ui/theme/dist/chunk-SBZTN3FG.mjs
'use client'

// src/anatomy/index.ts
import { anatomy } from "@chakra-ui/theme-tools";
var appShellAnatomy = anatomy("app-shell").parts(
  "container",
  "inner",
  "main"
);
var emptyStateAnatomy = anatomy("emptystate").parts(
  "container",
  "body",
  "icon",
  "title",
  "descripton",
  "actions",
  "footer"
);
var bannerAnatomy = anatomy("banner").parts(
  "container",
  "icon",
  "content",
  "title",
  "description",
  "actions",
  "close"
);
var hotkeysAnantomy = anatomy("hotkeys").parts(
  "container",
  "group",
  "groupTitle",
  "item",
  "command",
  "then"
);
var loadingOverlayAnatomy = anatomy("loading-overlay").parts(
  "overlay",
  "text"
);
var navGroupAnatomy = anatomy("nav-group").parts(
  "container",
  "title",
  "icon",
  "content"
);
var navItemAnatomy = anatomy("nav-item").parts(
  "item",
  "link",
  "inner",
  "icon",
  "label"
);
var nprogressAnatomy = anatomy("nprogress").parts("container", "bar");
var personaAnatomy = anatomy("persona").parts(
  "container",
  "details",
  "avatar",
  "label",
  "secondaryLabel",
  "tertiaryLabel"
);
var searchInputAnatomy = anatomy("search-input").parts(
  "input",
  "reset"
);
var sidebarAnatomy = anatomy("sidebar").parts(
  "container",
  "overlay",
  "section",
  "toggleWrapper",
  "toggle"
);
var stepperAnatomy = anatomy("stepper").parts(
  "container",
  "steps",
  "icon",
  "content",
  "title",
  "separator"
);
var structuredListAnatomy = anatomy("structured-list").parts(
  "list",
  "item",
  "button",
  "header",
  "cell",
  "icon"
);
var propertyAnatomy = anatomy("property").parts(
  "property",
  "label",
  "value"
);
var selectAnatomy = anatomy("select").parts(
  "addon",
  "field",
  "element"
);
var timelineAnatomy = anatomy("timeline").parts(
  "container",
  "item",
  "separator",
  "icon",
  "dot",
  "track",
  "content"
);

export {
  appShellAnatomy,
  emptyStateAnatomy,
  bannerAnatomy,
  hotkeysAnantomy,
  loadingOverlayAnatomy,
  navGroupAnatomy,
  navItemAnatomy,
  nprogressAnatomy,
  personaAnatomy,
  searchInputAnatomy,
  sidebarAnatomy,
  stepperAnatomy,
  structuredListAnatomy,
  propertyAnatomy,
  selectAnatomy,
  timelineAnatomy
};
