// @saas-ui/core/dist/index.js
'use client'
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  AppShell: () => AppShell,
  Banner: () => Banner,
  BannerActions: () => BannerActions,
  BannerCloseButton: () => BannerCloseButton,
  BannerContent: () => BannerContent,
  BannerDescription: () => BannerDescription,
  BannerIcon: () => BannerIcon,
  BannerTitle: () => BannerTitle,
  Br: () => Br,
  CalendarIcon: () => CalendarIcon,
  CheckIcon: () => CheckIcon2,
  ChevronDownIcon: () => ChevronDownIcon,
  ChevronLeftIcon: () => ChevronLeftIcon,
  ChevronRightIcon: () => ChevronRightIcon,
  ChevronUpIcon: () => ChevronUpIcon,
  CloseIcon: () => CloseIcon,
  CollapseProvider: () => CollapseProvider,
  Command: () => Command,
  ContextMenu: () => ContextMenu,
  ContextMenuItem: () => ContextMenuItem,
  ContextMenuList: () => ContextMenuList,
  ContextMenuProvider: () => ContextMenuProvider,
  ContextMenuTrigger: () => ContextMenuTrigger,
  EmptyState: () => EmptyState,
  EmptyStateActions: () => EmptyStateActions,
  EmptyStateBody: () => EmptyStateBody,
  EmptyStateContainer: () => EmptyStateContainer,
  EmptyStateDescription: () => EmptyStateDescription,
  EmptyStateFooter: () => EmptyStateFooter,
  EmptyStateIcon: () => EmptyStateIcon,
  EmptyStateTitle: () => EmptyStateTitle,
  ErrorBoundary: () => ErrorBoundary,
  FilterIcon: () => FilterIcon,
  HamburgerIcon: () => HamburgerIcon,
  IconBadge: () => IconBadge,
  Link: () => Link2,
  LoadingOverlay: () => LoadingOverlay,
  LoadingSpinner: () => LoadingSpinner,
  LoadingText: () => LoadingText,
  MinusIcon: () => MinusIcon,
  Nav: () => Nav,
  NavGroup: () => NavGroup,
  NavGroupContent: () => NavGroupContent,
  NavGroupStylesProvider: () => NavGroupStylesProvider,
  NavGroupTitle: () => NavGroupTitle,
  NavItem: () => NavItem,
  NavItemLabel: () => NavItemLabel,
  NavItemStylesProvider: () => NavItemStylesProvider,
  Navbar: () => Navbar,
  NavbarBrand: () => NavbarBrand,
  NavbarContent: () => NavbarContent,
  NavbarItem: () => NavbarItem,
  NavbarLink: () => NavbarLink,
  NavbarProvider: () => NavbarProvider,
  OverflowMenu: () => OverflowMenu,
  Persona: () => Persona,
  PersonaAvatar: () => PersonaAvatar,
  PersonaContainer: () => PersonaContainer,
  PersonaDetails: () => PersonaDetails,
  PersonaLabel: () => PersonaLabel,
  PersonaSecondaryLabel: () => PersonaSecondaryLabel,
  PersonaTertiaryLabel: () => PersonaTertiaryLabel,
  PlusIcon: () => PlusIcon,
  Presence: () => Presence,
  Property: () => Property,
  PropertyLabel: () => PropertyLabel,
  PropertyList: () => PropertyList,
  PropertyValue: () => PropertyValue,
  SaasContext: () => SaasContext,
  SaasProvider: () => SaasProvider,
  SearchIcon: () => SearchIcon,
  SearchInput: () => SearchInput,
  Sidebar: () => Sidebar,
  SidebarOverlay: () => SidebarOverlay,
  SidebarProvider: () => SidebarProvider,
  SidebarSection: () => SidebarSection,
  SidebarStylesProvider: () => SidebarStylesProvider,
  SidebarToggleButton: () => SidebarToggleButton,
  Snackbar: () => Snackbar,
  StepperProvider: () => StepperProvider,
  Steps: () => Steps,
  StepsCompleted: () => StepsCompleted,
  StepsContent: () => StepsContent,
  StepsItem: () => StepsItem,
  StructuredList: () => StructuredList,
  StructuredListButton: () => StructuredListButton,
  StructuredListCell: () => StructuredListCell,
  StructuredListHeader: () => StructuredListHeader,
  StructuredListIcon: () => StructuredListIcon,
  StructuredListItem: () => StructuredListItem,
  Timeline: () => Timeline,
  TimelineContent: () => TimelineContent,
  TimelineDot: () => TimelineDot,
  TimelineIcon: () => TimelineIcon,
  TimelineItem: () => TimelineItem,
  TimelineSeparator: () => TimelineSeparator,
  TimelineTrack: () => TimelineTrack,
  ViewIcon: () => ViewIcon,
  ViewOffIcon: () => ViewOffIcon,
  Web3Address: () => Web3Address,
  createIcon: () => createIcon2,
  createStandAloneSnackbar: () => createStandAloneSnackbar,
  defaultPresenceTokens: () => defaultPresenceTokens,
  splitProps: () => splitProps,
  useCollapse: () => useCollapse,
  useCollapseContext: () => useCollapseContext,
  useContextMenu: () => useContextMenu,
  useContextMenuContext: () => useContextMenuContext,
  useEmptyStateStyles: () => useEmptyStateStyles,
  useLink: () => useLink,
  useNavGroupStyles: () => useNavGroupStyles,
  useNavItemStyles: () => useNavItemStyles,
  useNavbar: () => useNavbar,
  useNavbarContext: () => useNavbarContext,
  useSaas: () => useSaas,
  useSidebarContext: () => useSidebarContext,
  useSidebarStyles: () => useSidebarStyles,
  useSidebarToggleButton: () => useSidebarToggleButton,
  useSnackbar: () => useSnackbar,
  useStep: () => useStep,
  useStepper: () => useStepper,
  useStepperContext: () => useStepperContext,
  useStepperNextButton: () => useStepperNextButton,
  useStepperPrevButton: () => useStepperPrevButton,
  useTimelineStyles: () => useTimelineStyles
});
module.exports = __toCommonJS(src_exports);

// src/command/command.tsx
var import_react = require("@chakra-ui/react");
var import_utils = require("@chakra-ui/utils");
var import_jsx_runtime = require("react/jsx-runtime");
var Key = ({ children }) => {
  if (typeof children !== "string") {
    return /* @__PURE__ */ (0, import_jsx_runtime.jsx)(import_jsx_runtime.Fragment, { children });
  }
  if (["then", "or", "+"].includes(children)) {
    return /* @__PURE__ */ (0, import_jsx_runtime.jsx)(import_react.chakra.span, { children });
  }
  return /* @__PURE__ */ (0, import_jsx_runtime.jsx)(import_react.Kbd, { children });
};
var Command = (props) => {
  const { children, className, ...rest } = props;
  if (typeof children !== "string") {
    return /* @__PURE__ */ (0, import_jsx_runtime.jsx)(import_jsx_runtime.Fragment, { children });
  }
  const keys = children.split(/\s|\+/);
  const styles = (0, import_react.useStyleConfig)("SuiCommand");
  const commandStyles = {
    "& > *:not(style) ~ *:not(style)": { marginEnd: 1 },
    color: "muted",
    ...styles
  };
  return /* @__PURE__ */ (0, import_jsx_runtime.jsx)(
    import_react.chakra.span,
    {
      className: (0, import_utils.cx)("sui-command", className),
      __css: commandStyles,
      ...rest,
      children: keys.map((key) => /* @__PURE__ */ (0, import_jsx_runtime.jsx)(Key, { children: key }, key))
    }
  );
};

// src/typography/br.tsx
var import_react2 = require("@chakra-ui/react");
var import_jsx_runtime2 = require("react/jsx-runtime");
var Br = (props) => {
  return /* @__PURE__ */ (0, import_jsx_runtime2.jsx)(import_react2.chakra.span, { ...props, children: /* @__PURE__ */ (0, import_jsx_runtime2.jsx)("br", {}) });
};

// src/empty-state/empty-state.tsx
var import_react3 = require("@chakra-ui/react");
var import_utils2 = require("@chakra-ui/utils");
var import_jsx_runtime3 = require("react/jsx-runtime");
var [StylesProvider, useStyles] = (0, import_react3.createStylesContext)("SuiEmptyState");
var useEmptyStateStyles = useStyles;
var EmptyState = (props) => {
  const {
    children,
    spacing,
    icon,
    title,
    description,
    actions,
    footer,
    ...rest
  } = props;
  return /* @__PURE__ */ (0, import_jsx_runtime3.jsx)(EmptyStateContainer, { ...rest, children: /* @__PURE__ */ (0, import_jsx_runtime3.jsxs)(EmptyStateBody, { spacing, children: [
    icon && /* @__PURE__ */ (0, import_jsx_runtime3.jsx)(EmptyStateIcon, { as: icon }),
    /* @__PURE__ */ (0, import_jsx_runtime3.jsxs)(import_react3.chakra.div, { children: [
      title && /* @__PURE__ */ (0, import_jsx_runtime3.jsx)(EmptyStateTitle, { children: title }),
      description && /* @__PURE__ */ (0, import_jsx_runtime3.jsx)(EmptyStateDescription, { children: description })
    ] }),
    children,
    actions && /* @__PURE__ */ (0, import_jsx_runtime3.jsx)(EmptyStateActions, { children: actions }),
    footer && /* @__PURE__ */ (0, import_jsx_runtime3.jsx)(EmptyStateFooter, { children: footer })
  ] }) });
};
EmptyState.displayName = "EmptyState";
var EmptyStateContainer = (props) => {
  const { children, spacing, ...rest } = props;
  const styles = (0, import_react3.useMultiStyleConfig)("SuiEmptyState", props);
  const containerProps = (0, import_react3.omitThemingProps)(rest);
  const containerStyles = {
    display: "flex",
    flex: 1,
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "center",
    ...styles.container
  };
  return /* @__PURE__ */ (0, import_jsx_runtime3.jsx)(StylesProvider, { value: styles, children: /* @__PURE__ */ (0, import_jsx_runtime3.jsx)(
    import_react3.chakra.div,
    {
      ...containerProps,
      __css: containerStyles,
      className: (0, import_utils2.cx)("sui-empty-state", props.className),
      children
    }
  ) });
};
EmptyStateContainer.displayName = "EmptyStateContainer";
var EmptyStateIcon = (0, import_react3.forwardRef)(
  (props, ref) => {
    const styles = useStyles();
    const iconStyles = {
      boxSize: 10,
      ...styles.icon
    };
    return /* @__PURE__ */ (0, import_jsx_runtime3.jsx)(
      import_react3.Icon,
      {
        ref,
        role: "presentation",
        sx: iconStyles,
        ...props,
        className: (0, import_utils2.cx)("sui-empty-state__icon", props.className)
      }
    );
  }
);
EmptyStateIcon.displayName = "EmptyStateIcon";
var EmptyStateTitle = (props) => {
  const styles = useStyles();
  const titleStyles = {
    mb: 1,
    ...styles.title
  };
  return /* @__PURE__ */ (0, import_jsx_runtime3.jsx)(
    import_react3.chakra.h3,
    {
      __css: titleStyles,
      ...props,
      className: (0, import_utils2.cx)("sui-empty-state__title", props.className)
    }
  );
};
EmptyStateTitle.displayName = "EmptyStateTitle";
var EmptyStateDescription = (props) => {
  const styles = useStyles();
  const descriptionStyles = {
    color: "muted",
    fontSize: "md",
    ...styles.description
  };
  return /* @__PURE__ */ (0, import_jsx_runtime3.jsx)(
    import_react3.chakra.p,
    {
      __css: descriptionStyles,
      ...props,
      className: (0, import_utils2.cx)("sui-empty-state__description", props.className)
    }
  );
};
EmptyStateDescription.displayName = "EmptyStateDescription";
var EmptyStateBody = (props) => {
  const { spacing = 8, ...rest } = props;
  const styles = useStyles();
  const bodyStyles = {
    ...styles.body
  };
  return /* @__PURE__ */ (0, import_jsx_runtime3.jsx)(
    import_react3.chakra.div,
    {
      __css: bodyStyles,
      ...rest,
      className: (0, import_utils2.cx)("sui-empty-state__body", props.className)
    }
  );
};
EmptyStateBody.displayName = "EmptyStateBody";
var EmptyStateActions = (props) => {
  const styles = useStyles();
  return /* @__PURE__ */ (0, import_jsx_runtime3.jsx)(
    import_react3.ButtonGroup,
    {
      sx: styles.actions,
      ...props,
      className: (0, import_utils2.cx)("sui-empty-state__actions", props.className)
    }
  );
};
EmptyStateActions.displayName = "EmptyStateActions";
var EmptyStateFooter = (props) => {
  const styles = useStyles();
  return /* @__PURE__ */ (0, import_jsx_runtime3.jsx)(
    import_react3.chakra.footer,
    {
      __css: styles.footer,
      ...props,
      className: (0, import_utils2.cx)("sui-empty-state__footer", props.className)
    }
  );
};
EmptyStateFooter.displayName = "EmptyStateFooter";

// src/error-boundary/error-boundary.tsx
var React2 = __toESM(require("react"));

// src/provider/saas-provider.tsx
var React = __toESM(require("react"));
var import_react4 = require("@chakra-ui/react");
var import_theme = require("@saas-ui/theme");
var import_jsx_runtime4 = require("react/jsx-runtime");
var SaasContext = React.createContext({});
function SaasProvider(props) {
  const { theme, linkComponent, onError, children, ...rest } = props;
  const context = {
    linkComponent,
    onError
  };
  return /* @__PURE__ */ (0, import_jsx_runtime4.jsx)(SaasContext.Provider, { value: context, children: /* @__PURE__ */ (0, import_jsx_runtime4.jsx)(import_react4.ChakraProvider, { ...rest, theme: theme || import_theme.theme, children }) });
}
var useSaas = () => React.useContext(SaasContext);

// src/provider/use-link.tsx
var import_react5 = require("@chakra-ui/react");
var import_jsx_runtime5 = require("react/jsx-runtime");
var Link = (props) => /* @__PURE__ */ (0, import_jsx_runtime5.jsx)(import_react5.chakra.a, { ...props });
function useLink() {
  const context = useSaas();
  if (context == null ? void 0 : context.linkComponent) {
    return context.linkComponent;
  }
  return Link;
}

// src/error-boundary/error-boundary.tsx
var import_jsx_runtime6 = require("react/jsx-runtime");
var ErrorBoundary = class extends React2.Component {
  constructor(props) {
    super(props);
    this.onError = (error2, errorInfo) => {
      var _a, _b, _c, _d;
      (_b = (_a = this.props).onError) == null ? void 0 : _b.call(_a, error2, errorInfo);
      (_d = (_c = this.context).onError) == null ? void 0 : _d.call(_c, error2, errorInfo);
    };
    this.state = { error: null };
  }
  static getDerivedStateFromError(error2) {
    return { error: error2 };
  }
  componentDidCatch(error2, errorInfo) {
    this.onError(error2, errorInfo);
  }
  render() {
    if (this.state.error) {
      return this.props.fallback || /* @__PURE__ */ (0, import_jsx_runtime6.jsx)("h1", { children: "Something went wrong." });
    }
    return this.props.children;
  }
};
ErrorBoundary.contextType = SaasContext;

// src/app-shell/app-shell.tsx
var React3 = __toESM(require("react"));
var import_react7 = require("@chakra-ui/react");
var import_utils3 = require("@chakra-ui/utils");

// src/app-shell/app-shell-context.ts
var import_react_utils = require("@chakra-ui/react-utils");
var import_react6 = require("@chakra-ui/react");

// src/sidebar/sidebar-utils.ts
var getBreakpoints = (breakpoint = "lg") => {
  return breakpoint ? {
    base: true,
    [breakpoint]: false
  } : { base: false };
};

// src/app-shell/app-shell-context.ts
var [AppShellProvider, useAppShellContext] = (0, import_react_utils.createContext)({
  strict: false,
  errorMessage: "AppShell context not available."
});
var useAppShell = (props) => {
  const disclosure = (0, import_react6.useDisclosure)();
  const breakpoints2 = getBreakpoints(props.toggleBreakpoint);
  const isMobile = (0, import_react6.useBreakpointValue)(breakpoints2, {
    fallback: props.toggleBreakpoint || "lg"
  });
  return {
    isSidebarOpen: disclosure.isOpen,
    closeSidebar: disclosure.onClose,
    openSidebar: disclosure.onOpen,
    toggleSidebar: disclosure.onToggle,
    isMobile
  };
};

// src/app-shell/app-shell.tsx
var import_jsx_runtime7 = require("react/jsx-runtime");
var [StylesProvider2, useStyles2] = (0, import_react7.createStylesContext)("SuiAppShell");
var AppShell = (0, import_react7.forwardRef)((props, ref) => {
  const styles = (0, import_react7.useMultiStyleConfig)("SuiAppShell", props);
  const {
    navbar,
    sidebar,
    aside,
    footer,
    children,
    mainRef,
    ...containerProps
  } = (0, import_react7.omitThemingProps)(props);
  const containerStyles = {
    flexDirection: "column",
    ...styles.container
  };
  const innerStyles = {
    flex: 1,
    minHeight: 0,
    // make sure child flex divs get correct height.
    minWidth: 0,
    // make sure child flex divs get correct width.
    ...styles.inner
  };
  const mainStyles = {
    flex: 1,
    flexDirection: "column",
    minWidth: 0,
    // make sure child flex divs get correct width.
    ...styles.main
  };
  const isSidebar = React3.isValidElement(sidebar) && sidebar.type.id === "Sidebar";
  const context = useAppShell({
    toggleBreakpoint: isSidebar ? sidebar == null ? void 0 : sidebar.props.toggleBreakpoint : void 0
  });
  return /* @__PURE__ */ (0, import_jsx_runtime7.jsx)(AppShellProvider, { value: context, children: /* @__PURE__ */ (0, import_jsx_runtime7.jsx)(StylesProvider2, { value: styles, children: /* @__PURE__ */ (0, import_jsx_runtime7.jsxs)(
    import_react7.Flex,
    {
      ref,
      ...containerProps,
      sx: containerStyles,
      className: (0, import_utils3.cx)("sui-app-shell", props.className),
      children: [
        navbar,
        /* @__PURE__ */ (0, import_jsx_runtime7.jsxs)(import_react7.Flex, { sx: innerStyles, className: "saas-app-shell__inner", children: [
          sidebar,
          /* @__PURE__ */ (0, import_jsx_runtime7.jsx)(
            import_react7.Flex,
            {
              ref: mainRef,
              sx: mainStyles,
              className: "saas-app-shell__main",
              children
            }
          ),
          aside
        ] }),
        footer
      ]
    }
  ) }) });
});
AppShell.displayName = "AppShell";

// src/banner/banner.tsx
var import_react11 = require("@chakra-ui/react");
var import_utils4 = require("@chakra-ui/utils");
var import_react_utils2 = require("@chakra-ui/react-utils");

// src/banner/icons.tsx
var import_react8 = require("@chakra-ui/react");
var import_jsx_runtime8 = require("react/jsx-runtime");
function CheckIcon(props) {
  return /* @__PURE__ */ (0, import_jsx_runtime8.jsx)(import_react8.Icon, { viewBox: "0 0 24 24", ...props, children: /* @__PURE__ */ (0, import_jsx_runtime8.jsx)(
    "path",
    {
      fill: "currentColor",
      d: "M12,0A12,12,0,1,0,24,12,12.014,12.014,0,0,0,12,0Zm6.927,8.2-6.845,9.289a1.011,1.011,0,0,1-1.43.188L5.764,13.769a1,1,0,1,1,1.25-1.562l4.076,3.261,6.227-8.451A1,1,0,1,1,18.927,8.2Z"
    }
  ) });
}
function InfoIcon(props) {
  return /* @__PURE__ */ (0, import_jsx_runtime8.jsx)(import_react8.Icon, { viewBox: "0 0 24 24", ...props, children: /* @__PURE__ */ (0, import_jsx_runtime8.jsx)(
    "path",
    {
      fill: "currentColor",
      d: "M12,0A12,12,0,1,0,24,12,12.013,12.013,0,0,0,12,0Zm.25,5a1.5,1.5,0,1,1-1.5,1.5A1.5,1.5,0,0,1,12.25,5ZM14.5,18.5h-4a1,1,0,0,1,0-2h.75a.25.25,0,0,0,.25-.25v-4.5a.25.25,0,0,0-.25-.25H10.5a1,1,0,0,1,0-2h1a2,2,0,0,1,2,2v4.75a.25.25,0,0,0,.25.25h.75a1,1,0,1,1,0,2Z"
    }
  ) });
}
function WarningIcon(props) {
  return /* @__PURE__ */ (0, import_jsx_runtime8.jsx)(import_react8.Icon, { viewBox: "0 0 24 24", ...props, children: /* @__PURE__ */ (0, import_jsx_runtime8.jsx)(
    "path",
    {
      fill: "currentColor",
      d: "M11.983,0a12.206,12.206,0,0,0-8.51,3.653A11.8,11.8,0,0,0,0,12.207,11.779,11.779,0,0,0,11.8,24h.214A12.111,12.111,0,0,0,24,11.791h0A11.766,11.766,0,0,0,11.983,0ZM10.5,16.542a1.476,1.476,0,0,1,1.449-1.53h.027a1.527,1.527,0,0,1,1.523,1.47,1.475,1.475,0,0,1-1.449,1.53h-.027A1.529,1.529,0,0,1,10.5,16.542ZM11,12.5v-6a1,1,0,0,1,2,0v6a1,1,0,1,1-2,0Z"
    }
  ) });
}

// src/banner/banner.tsx
var import_framer_motion4 = require("framer-motion");

// src/banner/banner-transition.tsx
var React4 = __toESM(require("react"));
var import_system = require("@chakra-ui/system");

// ../../node_modules/@chakra-ui/transition/dist/chunk-6NHXDBFO.mjs
var TRANSITION_EASINGS = {
  ease: [0.25, 0.1, 0.25, 1],
  easeIn: [0.4, 0, 1, 1],
  easeOut: [0, 0, 0.2, 1],
  easeInOut: [0.4, 0, 0.2, 1]
};
var TRANSITION_DEFAULTS = {
  enter: {
    duration: 0.2,
    ease: TRANSITION_EASINGS.easeOut
  },
  exit: {
    duration: 0.1,
    ease: TRANSITION_EASINGS.easeIn
  }
};
var withDelay = {
  enter: (transition, delay) => ({
    ...transition,
    delay: typeof delay === "number" ? delay : delay == null ? void 0 : delay["enter"]
  }),
  exit: (transition, delay) => ({
    ...transition,
    delay: typeof delay === "number" ? delay : delay == null ? void 0 : delay["exit"]
  })
};

// ../../node_modules/@chakra-ui/shared-utils/dist/index.mjs
var cx4 = (...classNames) => classNames.filter(Boolean).join(" ");

// ../../node_modules/@chakra-ui/transition/dist/chunk-SR4VX3RF.mjs
var import_framer_motion = require("framer-motion");
var import_react9 = require("react");
var import_jsx_runtime9 = require("react/jsx-runtime");
var variants = {
  exit: ({ reverse, initialScale, transition, transitionEnd, delay }) => {
    var _a;
    return {
      opacity: 0,
      ...reverse ? { scale: initialScale, transitionEnd: transitionEnd == null ? void 0 : transitionEnd.exit } : { transitionEnd: { scale: initialScale, ...transitionEnd == null ? void 0 : transitionEnd.exit } },
      transition: (_a = transition == null ? void 0 : transition.exit) != null ? _a : withDelay.exit(TRANSITION_DEFAULTS.exit, delay)
    };
  },
  enter: ({ transitionEnd, transition, delay }) => {
    var _a;
    return {
      opacity: 1,
      scale: 1,
      transition: (_a = transition == null ? void 0 : transition.enter) != null ? _a : withDelay.enter(TRANSITION_DEFAULTS.enter, delay),
      transitionEnd: transitionEnd == null ? void 0 : transitionEnd.enter
    };
  }
};
var scaleFadeConfig = {
  initial: "exit",
  animate: "enter",
  exit: "exit",
  variants
};
var ScaleFade = (0, import_react9.forwardRef)(
  function ScaleFade2(props, ref) {
    const {
      unmountOnExit,
      in: isOpen,
      reverse = true,
      initialScale = 0.95,
      className,
      transition,
      transitionEnd,
      delay,
      ...rest
    } = props;
    const show = unmountOnExit ? isOpen && unmountOnExit : true;
    const animate = isOpen || unmountOnExit ? "enter" : "exit";
    const custom = { initialScale, reverse, transition, transitionEnd, delay };
    return /* @__PURE__ */ (0, import_jsx_runtime9.jsx)(import_framer_motion.AnimatePresence, { custom, children: show && /* @__PURE__ */ (0, import_jsx_runtime9.jsx)(
      import_framer_motion.motion.div,
      {
        ref,
        className: cx4("chakra-offset-slide", className),
        ...scaleFadeConfig,
        animate,
        custom,
        ...rest
      }
    ) });
  }
);
ScaleFade.displayName = "ScaleFade";

// ../../node_modules/@chakra-ui/transition/dist/chunk-7254PCTD.mjs
var import_framer_motion2 = require("framer-motion");
var import_react10 = require("react");
var import_jsx_runtime10 = require("react/jsx-runtime");
var variants2 = {
  initial: ({ offsetX, offsetY, transition, transitionEnd, delay }) => {
    var _a;
    return {
      opacity: 0,
      x: offsetX,
      y: offsetY,
      transition: (_a = transition == null ? void 0 : transition.exit) != null ? _a : withDelay.exit(TRANSITION_DEFAULTS.exit, delay),
      transitionEnd: transitionEnd == null ? void 0 : transitionEnd.exit
    };
  },
  enter: ({ transition, transitionEnd, delay }) => {
    var _a;
    return {
      opacity: 1,
      x: 0,
      y: 0,
      transition: (_a = transition == null ? void 0 : transition.enter) != null ? _a : withDelay.enter(TRANSITION_DEFAULTS.enter, delay),
      transitionEnd: transitionEnd == null ? void 0 : transitionEnd.enter
    };
  },
  exit: ({ offsetY, offsetX, transition, transitionEnd, reverse, delay }) => {
    var _a;
    const offset2 = { x: offsetX, y: offsetY };
    return {
      opacity: 0,
      transition: (_a = transition == null ? void 0 : transition.exit) != null ? _a : withDelay.exit(TRANSITION_DEFAULTS.exit, delay),
      ...reverse ? { ...offset2, transitionEnd: transitionEnd == null ? void 0 : transitionEnd.exit } : { transitionEnd: { ...offset2, ...transitionEnd == null ? void 0 : transitionEnd.exit } }
    };
  }
};
var slideFadeConfig = {
  initial: "initial",
  animate: "enter",
  exit: "exit",
  variants: variants2
};
var SlideFade = (0, import_react10.forwardRef)(
  function SlideFade2(props, ref) {
    const {
      unmountOnExit,
      in: isOpen,
      reverse = true,
      className,
      offsetX = 0,
      offsetY = 8,
      transition,
      transitionEnd,
      delay,
      ...rest
    } = props;
    const show = unmountOnExit ? isOpen && unmountOnExit : true;
    const animate = isOpen || unmountOnExit ? "enter" : "exit";
    const custom = {
      offsetX,
      offsetY,
      reverse,
      transition,
      transitionEnd,
      delay
    };
    return /* @__PURE__ */ (0, import_jsx_runtime10.jsx)(import_framer_motion2.AnimatePresence, { custom, children: show && /* @__PURE__ */ (0, import_jsx_runtime10.jsx)(
      import_framer_motion2.motion.div,
      {
        ref,
        className: cx4("chakra-offset-slide", className),
        custom,
        ...slideFadeConfig,
        animate,
        ...rest
      }
    ) });
  }
);
SlideFade.displayName = "SlideFade";

// src/banner/banner-transition.tsx
var import_framer_motion3 = require("framer-motion");
var import_jsx_runtime11 = require("react/jsx-runtime");
var defaultTransition = {
  enter: {
    duration: 0.2,
    ease: TRANSITION_EASINGS.easeOut
  },
  exit: {
    duration: 0.2,
    ease: TRANSITION_EASINGS.easeIn
  }
};
var transitions = {
  slideOutTop: {
    ...slideFadeConfig,
    custom: {
      offsetY: "-100%",
      reverse: true,
      transition: defaultTransition
    },
    initial: "enter"
  },
  slideOutBottom: {
    ...slideFadeConfig,
    custom: {
      offsetY: "100%",
      reverse: true,
      transition: defaultTransition
    },
    initial: "enter"
  },
  fade: {
    ...slideFadeConfig,
    custom: {
      transition: defaultTransition
    },
    initial: "enter"
  },
  scale: {
    ...scaleFadeConfig,
    custom: { initialScale: 0.1, reverse: true, transition: defaultTransition },
    initial: "enter"
  },
  none: {
    custom: {}
  }
};
var Motion = (0, import_system.chakra)(import_framer_motion3.motion.div);
var BannerTransition = React4.forwardRef(
  (props, ref) => {
    const { motionPreset, ...rest } = props;
    const preset = transitions[motionPreset];
    const motionProps = {
      ...preset
    };
    return /* @__PURE__ */ (0, import_jsx_runtime11.jsx)(Motion, { ref, ...motionProps, ...rest });
  }
);

// src/banner/banner.tsx
var import_jsx_runtime12 = require("react/jsx-runtime");
var [StylesProvider3, useStyles3] = (0, import_react11.createStylesContext)("SuiBanner");
var STATUSES = {
  info: { icon: InfoIcon, colorScheme: "blue" },
  warning: { icon: WarningIcon, colorScheme: "orange" },
  success: { icon: CheckIcon, colorScheme: "green" },
  error: { icon: WarningIcon, colorScheme: "red" }
};
var [BannerProvider, useBannerContext] = (0, import_react_utils2.createContext)({
  name: "BannerContext",
  errorMessage: "useBannerContext: `context` is undefined. Seems you forgot to wrap banner components in `<Banner />`"
});
var Banner = (0, import_react11.forwardRef)((props, ref) => {
  var _a;
  const {
    id,
    status = "info",
    isOpen = true,
    onClose,
    motionPreset = "slideOutTop",
    ...rest
  } = (0, import_react11.omitThemingProps)(props);
  const colorScheme = (_a = props.colorScheme) != null ? _a : STATUSES[status].colorScheme;
  const styles = (0, import_react11.useMultiStyleConfig)("SuiBanner", {
    ...props,
    colorScheme
  });
  const containerStyles = {
    width: "100%",
    display: "flex",
    alignItems: "center",
    position: "relative",
    overflow: "hidden",
    ...styles.container
  };
  const context = {
    id: id || `banner-${(0, import_react11.useId)()}`,
    status,
    onClose,
    isOpen
  };
  const role = ["warning", "error"].includes(status) ? "alert" : "status";
  const animate = isOpen ? "enter" : "exit";
  return /* @__PURE__ */ (0, import_jsx_runtime12.jsx)(BannerProvider, { value: context, children: /* @__PURE__ */ (0, import_jsx_runtime12.jsx)(StylesProvider3, { value: styles, children: /* @__PURE__ */ (0, import_jsx_runtime12.jsx)(import_framer_motion4.AnimatePresence, { children: isOpen && /* @__PURE__ */ (0, import_jsx_runtime12.jsx)(
    BannerTransition,
    {
      id: context.id,
      role,
      ref,
      motionPreset,
      animate,
      ...rest,
      className: (0, import_utils4.cx)("sui-banner", props.className),
      __css: containerStyles
    }
  ) }) }) });
});
Banner.displayName = "Banner";
var BannerContent = (0, import_react11.forwardRef)(
  (props, ref) => {
    const styles = useStyles3();
    return /* @__PURE__ */ (0, import_jsx_runtime12.jsx)(
      import_react11.chakra.div,
      {
        ref,
        ...props,
        className: (0, import_utils4.cx)("sui-banner__content", props.className),
        __css: styles.content
      }
    );
  }
);
BannerContent.displayName = "BannerContent";
var BannerTitle = (0, import_react11.forwardRef)((props, ref) => {
  const styles = useStyles3();
  return /* @__PURE__ */ (0, import_jsx_runtime12.jsx)(
    import_react11.chakra.div,
    {
      ref,
      ...props,
      className: (0, import_utils4.cx)("sui-banner__title", props.className),
      __css: styles.title
    }
  );
});
BannerTitle.displayName = "BannerTitle";
var BannerDescription = (0, import_react11.forwardRef)(
  (props, ref) => {
    const styles = useStyles3();
    const descriptionStyles = {
      display: "inline",
      ...styles.description
    };
    return /* @__PURE__ */ (0, import_jsx_runtime12.jsx)(
      import_react11.chakra.div,
      {
        ref,
        ...props,
        className: (0, import_utils4.cx)("sui-banner__desc", props.className),
        __css: descriptionStyles
      }
    );
  }
);
BannerDescription.displayName = "BannerDescription";
var BannerActions = (0, import_react11.forwardRef)(
  (props, ref) => {
    const { children, variant } = props;
    const styles = useStyles3();
    return /* @__PURE__ */ (0, import_jsx_runtime12.jsx)(
      import_react11.chakra.div,
      {
        ref,
        ...props,
        className: (0, import_utils4.cx)("sui-banner__actions", props.className),
        __css: styles.actions,
        children: /* @__PURE__ */ (0, import_jsx_runtime12.jsx)(import_react11.ButtonGroup, { variant, children })
      }
    );
  }
);
BannerActions.displayName = "BannerActions";
var BannerIcon = (props) => {
  const { status } = useBannerContext();
  const { icon, ...rest } = props;
  const { icon: statusIcon } = STATUSES[status];
  const styles = useStyles3();
  return /* @__PURE__ */ (0, import_jsx_runtime12.jsx)(
    import_react11.chakra.span,
    {
      display: "inherit",
      ...rest,
      className: (0, import_utils4.cx)("sui-banner__icon", props.className),
      __css: styles.icon,
      children: /* @__PURE__ */ (0, import_jsx_runtime12.jsx)(import_react11.Icon, { as: icon || statusIcon, w: "100%", h: "100%" })
    }
  );
};
BannerIcon.displayName = "BannerIcon";
var BannerCloseButton = (0, import_react11.forwardRef)(
  (props, ref) => {
    const { onClick, className, ...rest } = props;
    const { onClose, isOpen, id } = useBannerContext();
    const _className = (0, import_utils4.cx)("sui-banner__close-btn", className);
    const styles = useStyles3();
    return /* @__PURE__ */ (0, import_jsx_runtime12.jsx)(
      import_react11.CloseButton,
      {
        ref,
        __css: styles.closeButton,
        className: _className,
        onClick: (0, import_utils4.callAllHandlers)(onClick, (event) => {
          event.stopPropagation();
          onClose == null ? void 0 : onClose();
        }),
        "aria-controls": id,
        "aria-expanded": (isOpen == null ? void 0 : isOpen.toString()) ? "true" : "false",
        ...rest
      }
    );
  }
);
BannerCloseButton.displayName = "BannerCloseButton";

// src/collapse/use-collapse.ts
var React5 = __toESM(require("react"));
var import_react_utils3 = require("@chakra-ui/react-utils");
var import_react12 = require("@chakra-ui/react");
var import_utils5 = require("@chakra-ui/utils");
var [CollapseProvider, useCollapseContext] = (0, import_react_utils3.createContext)({
  name: "UseCollapseReturn"
});
var useCollapse = (props = {}) => {
  const { isCollapsible = true, ...rest } = props;
  const id = `collapse-${(0, import_react12.useId)()}`;
  const { isOpen, onToggle, onOpen, onClose } = (0, import_react12.useDisclosure)(rest);
  const getToggleProps = React5.useCallback(
    (props2 = {}) => {
      const { className, ...rest2 } = props2;
      if (isCollapsible) {
        return {
          className: (0, import_utils5.cx)("sui-collapse-toggle", className),
          onClick: () => onToggle(),
          "aria-expanded": isOpen.toString(),
          "aria-controls": id,
          ...rest2
        };
      }
      return {};
    },
    [isCollapsible, isOpen]
  );
  const getCollapseProps = React5.useCallback(
    (props2 = {}) => {
      return {
        id,
        in: isOpen,
        ...props2
      };
    },
    [isOpen]
  );
  return {
    isCollapsible,
    isOpen,
    getToggleProps,
    getCollapseProps,
    onToggle,
    onOpen,
    onClose
  };
};

// src/structured-list/list.tsx
var React6 = __toESM(require("react"));
var import_react13 = require("@chakra-ui/react");
var import_utils6 = require("@chakra-ui/utils");
var import_react_utils4 = require("@chakra-ui/react-utils");
var import_dom_utils = require("@zag-js/dom-utils");
var import_jsx_runtime13 = require("react/jsx-runtime");
var import_react14 = require("react");
var [StylesProvider4, useStyles4] = (0, import_react13.createStylesContext)("SuiStructuredList");
var [StructuredListProvider, useStructuredListContext] = (0, import_react_utils4.createContext)({
  name: "StructuredListContext",
  errorMessage: "useStructuredListContext: `context` is undefined. Seems you forgot to wrap the components in `<StructuredList />`"
});
function queryAllItems(root) {
  return (0, import_dom_utils.queryAll)(root, `[role='button']:not([disabled])`);
}
var useStructuredList = (props) => {
  var _a;
  const id = (0, import_react13.useId)();
  const ref = React6.useRef(null);
  const [focusId, setFocusId] = React6.useState(null);
  const listProps = {
    onBlur: (0, import_utils6.callAllHandlers)(props.onBlur, (e) => {
      if (e.relatedTarget) {
        const items = queryAllItems(ref.current);
        if (!items.includes(e.relatedTarget)) {
          setFocusId(null);
        }
      }
    })
  };
  return {
    id: (_a = props.id) != null ? _a : id,
    containerRef: ref,
    focusId,
    setFocusId,
    listProps
  };
};
var StructuredList = (0, import_react13.forwardRef)(
  (props, ref) => {
    const { items, children, ...rest } = props;
    const styles = (0, import_react13.useMultiStyleConfig)("SuiStructuredList", rest);
    const ownProps = (0, import_react13.omitThemingProps)(rest);
    let content;
    if (items) {
      content = items.map((item, i) => {
        return /* @__PURE__ */ (0, import_react14.createElement)(StructuredListItem, { ...item, key: item.id || i });
      });
    } else {
      content = children;
    }
    const listStyles = {
      py: 2,
      position: "relative",
      ...styles.list
    };
    const { listProps, ...context } = useStructuredList(props);
    return /* @__PURE__ */ (0, import_jsx_runtime13.jsx)(StructuredListProvider, { value: context, children: /* @__PURE__ */ (0, import_jsx_runtime13.jsx)(StylesProvider4, { value: styles, children: /* @__PURE__ */ (0, import_jsx_runtime13.jsx)(
      import_react13.chakra.ul,
      {
        ref: (0, import_react13.useMergeRefs)(ref, context.containerRef),
        __css: listStyles,
        ...ownProps,
        ...listProps,
        className: (0, import_utils6.cx)("sui-list", props.className),
        children: content
      }
    ) }) });
  }
);
StructuredList.displayName = "StructuredList";
var StructuredListHeader = (0, import_react13.forwardRef)(
  (props, ref) => {
    const {
      children,
      onClick,
      action,
      role = "heading",
      level = 1,
      ...rest
    } = props;
    const styles = useStyles4();
    return /* @__PURE__ */ (0, import_jsx_runtime13.jsxs)(
      import_react13.chakra.li,
      {
        ref,
        __css: styles.header,
        onClick,
        ...rest,
        className: (0, import_utils6.cx)("sui-list__header", props.className),
        children: [
          /* @__PURE__ */ (0, import_jsx_runtime13.jsx)(import_react13.chakra.span, { flex: "1", userSelect: "none", role, "aria-level": level, children }),
          action
        ]
      }
    );
  }
);
StructuredListHeader.displayName = "StructuredListHeader";
var StructuredListItem = (0, import_react13.forwardRef)(
  (props, ref) => {
    const { onClick, href, as, children, isDisabled: isDisabled2, ...rest } = props;
    const styles = useStyles4();
    const isButton = !!(onClick || href);
    const ContentWrapper = isButton ? StructuredListButton : React6.Fragment;
    const disablePadding = !!isButton;
    const itemStyles = {
      ...styles.item,
      ...disablePadding ? { py: 0, px: 0 } : {}
    };
    const wrapperProps = isButton ? {
      onClick,
      href,
      as,
      isDisabled: isDisabled2
    } : {};
    const content = isButton ? /* @__PURE__ */ (0, import_jsx_runtime13.jsx)(ContentWrapper, { ...wrapperProps, children }) : children;
    return /* @__PURE__ */ (0, import_jsx_runtime13.jsx)(
      import_react13.chakra.li,
      {
        ref,
        __css: itemStyles,
        ...rest,
        className: (0, import_utils6.cx)("sui-list__item", props.className),
        children: content
      }
    );
  }
);
StructuredListItem.displayName = "StructuredListItem";
var useStructuredListButton = (props) => {
  var _a;
  const {
    id: containerId,
    containerRef,
    focusId,
    setFocusId
  } = useStructuredListContext();
  const id = `${containerId}-${(0, import_react13.useId)()}`;
  const buttonId = (_a = props.id) != null ? _a : id;
  const isFocused = focusId === buttonId;
  function getItems() {
    return (0, import_dom_utils.queryAll)(
      containerRef.current,
      `.sui-list__item-button:not([aria-disabled=true])`
    );
  }
  const buttonProps = {
    id: buttonId,
    ["data-focus"]: (0, import_utils6.dataAttr)(isFocused),
    ["aria-disabled"]: props.isDisabled ? "true" : void 0,
    tabIndex: props.isDisabled ? -1 : 0,
    onFocus: (0, import_utils6.callAllHandlers)(props.onFocus, () => {
      setFocusId(buttonId);
    }),
    onKeyDown: (0, import_utils6.callAllHandlers)(
      props.onKeyDown,
      React6.useCallback(
        (e) => {
          const items = getItems();
          const keyMap = {
            ArrowUp: () => {
              var _a2;
              (_a2 = (0, import_dom_utils.prevById)(items, buttonId)) == null ? void 0 : _a2.focus();
            },
            ArrowDown: () => {
              var _a2;
              (_a2 = (0, import_dom_utils.nextById)(items, buttonId)) == null ? void 0 : _a2.focus();
            },
            Home: () => {
              var _a2;
              (_a2 = items[0]) == null ? void 0 : _a2.focus();
            },
            End: () => {
              var _a2;
              (_a2 = items[items.length - 1]) == null ? void 0 : _a2.focus();
            }
          };
          if (keyMap[e.key]) {
            e.preventDefault();
            keyMap[e.key](e);
          }
        },
        [buttonId]
      )
    ),
    onClick: (e) => {
      var _a2;
      if (props.isDisabled) {
        e.preventDefault();
        e.stopPropagation();
        return;
      }
      (_a2 = props.onClick) == null ? void 0 : _a2.call(props, e);
    }
  };
  return {
    buttonProps
  };
};
var StructuredListButton = (0, import_react13.forwardRef)((props, ref) => {
  const { children, isDisabled: isDisabled2, ...rest } = props;
  const { buttonProps } = useStructuredListButton(props);
  const styles = useStyles4();
  return /* @__PURE__ */ (0, import_jsx_runtime13.jsx)(
    import_react13.chakra.div,
    {
      ref,
      __css: styles.button,
      role: "button",
      ...rest,
      ...buttonProps,
      className: (0, import_utils6.cx)("sui-list__item-button", props.className),
      children
    }
  );
});
StructuredListButton.displayName = "StructuredListButton";
var StructuredListIcon = (props) => {
  const { children, spacing, size = 5, as, ...rest } = props;
  const styles = useStyles4();
  const iconStyles = {
    ...styles.icon,
    me: spacing
  };
  let _icon = children;
  if (!_icon && as) {
    _icon = /* @__PURE__ */ (0, import_jsx_runtime13.jsx)(import_react13.Icon, { as, role: "presentation", boxSize: size });
  } else if (React6.isValidElement(_icon) && _icon.type === import_react13.Icon) {
    _icon = React6.cloneElement(_icon, {
      boxSize: size
    });
  }
  return /* @__PURE__ */ (0, import_jsx_runtime13.jsx)(
    import_react13.chakra.div,
    {
      __css: iconStyles,
      ...rest,
      className: (0, import_utils6.cx)("sui-list__item-icon", props.className),
      children: _icon
    }
  );
};
StructuredListIcon.displayName = "StructuredListItemIcon";
var StructuredListCell = (props) => {
  const { children, ...rest } = props;
  const styles = useStyles4();
  return /* @__PURE__ */ (0, import_jsx_runtime13.jsx)(
    import_react13.chakra.div,
    {
      __css: styles.cell,
      ...rest,
      className: (0, import_utils6.cx)("sui-list__cell", props.className),
      children
    }
  );
};
StructuredListCell.displayName = "StructuredListCell";

// src/link/link.tsx
var import_react15 = require("@chakra-ui/react");
var import_jsx_runtime14 = require("react/jsx-runtime");
var Link2 = (0, import_react15.forwardRef)((props, ref) => {
  const LinkComponent = useLink();
  const { href, ...rest } = props;
  return /* @__PURE__ */ (0, import_jsx_runtime14.jsx)(import_react15.Link, { as: LinkComponent, ref, href, ...rest });
});
Link2.displayName = "Link";

// src/loading-overlay/loading-overlay.tsx
var import_react16 = require("react");
var import_react17 = require("@chakra-ui/react");
var import_utils7 = require("@chakra-ui/utils");
var import_framer_motion5 = require("framer-motion");
var import_jsx_runtime15 = require("react/jsx-runtime");
var [StylesProvider5, useStyles5] = (0, import_react17.createStylesContext)("SuiLoadingOverlay");
var Motion2 = (0, import_react17.chakra)(import_framer_motion5.motion.div);
var LoadingOverlay = (props) => {
  const styles = (0, import_react17.useMultiStyleConfig)("SuiLoadingOverlay", props);
  const {
    children,
    isLoading = true,
    spacing = 2,
    motionPreset,
    ...rest
  } = (0, import_react17.omitThemingProps)(props);
  const overlayStyles = {
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "center",
    "& > *:not(style) ~ *:not(style)": { marginTop: spacing },
    ...styles.overlay
  };
  const [animateInitial] = (0, import_react16.useState)(!isLoading && motionPreset !== "none");
  const motionProps = motionPreset === "none" ? {} : import_react17.fadeConfig;
  return /* @__PURE__ */ (0, import_jsx_runtime15.jsx)(StylesProvider5, { value: styles, children: /* @__PURE__ */ (0, import_jsx_runtime15.jsx)(import_framer_motion5.AnimatePresence, { initial: animateInitial, children: isLoading && /* @__PURE__ */ (0, import_jsx_runtime15.jsx)(
    Motion2,
    {
      ...motionProps,
      ...rest,
      __css: overlayStyles,
      className: (0, import_utils7.cx)("chakra-loading-overlay", props.className),
      children
    }
  ) }) });
};
LoadingOverlay.displayName = "LoadingOverlay";
var LoadingSpinner = import_react17.Spinner;
var LoadingText = (props) => {
  const styles = useStyles5();
  return /* @__PURE__ */ (0, import_jsx_runtime15.jsx)(import_react17.chakra.p, { ...props, __css: styles.text });
};

// src/menu/overflow-menu.tsx
var import_react19 = require("@chakra-ui/react");

// src/menu/icons.tsx
var import_react18 = require("@chakra-ui/react");
var import_jsx_runtime16 = require("react/jsx-runtime");
var MoreIcon = (props) => {
  return /* @__PURE__ */ (0, import_jsx_runtime16.jsxs)(import_react18.Icon, { viewBox: "0 0 24 24", strokeWidth: "2", stroke: "currentColor", ...props, children: [
    /* @__PURE__ */ (0, import_jsx_runtime16.jsx)("circle", { cx: "12", cy: "12", r: "1" }),
    /* @__PURE__ */ (0, import_jsx_runtime16.jsx)("circle", { cx: "12", cy: "5", r: "1" }),
    /* @__PURE__ */ (0, import_jsx_runtime16.jsx)("circle", { cx: "12", cy: "19", r: "1" })
  ] });
};

// src/menu/overflow-menu.tsx
var import_jsx_runtime17 = require("react/jsx-runtime");
var OverflowMenu = (props) => {
  const {
    variant = "ghost",
    size,
    label = "More",
    icon = /* @__PURE__ */ (0, import_jsx_runtime17.jsx)(MoreIcon, {}),
    menuListProps,
    children,
    ...rest
  } = props;
  return /* @__PURE__ */ (0, import_jsx_runtime17.jsxs)(import_react19.Menu, { ...rest, children: [
    /* @__PURE__ */ (0, import_jsx_runtime17.jsx)(
      import_react19.MenuButton,
      {
        as: import_react19.IconButton,
        icon,
        "aria-label": label,
        variant,
        size
      }
    ),
    /* @__PURE__ */ (0, import_jsx_runtime17.jsx)(import_react19.MenuList, { ...menuListProps, children })
  ] });
};
OverflowMenu.displayName = "OverflowMenu";

// src/menu/context-menu.tsx
var React8 = __toESM(require("react"));
var import_react20 = require("react");
var import_react21 = require("@chakra-ui/react");
var import_react_utils5 = require("@chakra-ui/react-utils");
var import_utils8 = require("@chakra-ui/utils");
var import_interactions = require("@react-aria/interactions");
var import_dom_event = require("@zag-js/dom-event");
var import_jsx_runtime18 = require("react/jsx-runtime");
var [ContextMenuProvider, useContextMenuContext] = (0, import_react_utils5.createContext)({
  name: "UseContextMenuContext",
  strict: false
});
var useContextMenu = (props) => {
  const { closeOnBlur = true } = props;
  const [isOpen, setIsOpen] = (0, import_react20.useState)(false);
  const [anchor, setAnchor] = (0, import_react20.useState)({ x: 0, y: 0 });
  const triggerRef = (0, import_react20.useRef)(null);
  const menuRef = (0, import_react20.useRef)(null);
  (0, import_react21.useEventListener)("contextmenu", (e) => {
    var _a;
    if (!((_a = triggerRef.current) == null ? void 0 : _a.contains(e.target)) && e.target !== triggerRef.current) {
      setIsOpen(false);
    } else {
      e.preventDefault();
      e.stopPropagation();
    }
  });
  (0, import_react21.useOutsideClick)({
    enabled: isOpen && closeOnBlur,
    ref: menuRef,
    handler: (event) => {
      var _a, _b;
      if (!(event.button === 2 && ((_a = triggerRef.current) == null ? void 0 : _a.contains(event.target))) && ((_b = menuRef.current) == null ? void 0 : _b.parentElement) !== event.target) {
        onClose();
      }
    }
  });
  const onOpen = (0, import_react20.useCallback)((event) => {
    const point = (0, import_dom_event.getEventPoint)(event);
    setAnchor(point);
    setIsOpen(true);
  }, []);
  const onClose = (0, import_react20.useCallback)(() => {
    var _a;
    (_a = props.onClose) == null ? void 0 : _a.call(props);
    setIsOpen(false);
  }, [props.onClose, setIsOpen]);
  return {
    isOpen,
    anchor,
    triggerRef,
    menuRef,
    onClose,
    onOpen
  };
};
var ContextMenu = (props) => {
  const { children, ...rest } = props;
  const ctx = useContextMenu(props);
  const context = React8.useMemo(() => ctx, [ctx]);
  const { isOpen, onClose } = context;
  return /* @__PURE__ */ (0, import_jsx_runtime18.jsx)(
    import_react21.Menu,
    {
      gutter: 0,
      ...rest,
      isOpen,
      onClose,
      closeOnBlur: false,
      children: (fnProps) => /* @__PURE__ */ (0, import_jsx_runtime18.jsx)(ContextMenuProvider, { value: context, children: (0, import_utils8.runIfFn)(children, fnProps) })
    }
  );
};
ContextMenu.displayName = "ContextMenu";
var generateClientRect = (x = 0, y = 0) => {
  return () => {
    return {
      width: 0,
      height: 0,
      top: y,
      left: x,
      right: x,
      bottom: y
    };
  };
};
var isTouchDevice = () => {
  return typeof window !== void 0 && window.matchMedia("(hover: none)").matches;
};
var useContextMenuTrigger = (props, ref) => {
  const { triggerRef, onOpen, onClose, anchor } = useContextMenuContext();
  const menu = (0, import_react21.useMenuContext)();
  const { popper, openAndFocusFirstItem } = menu;
  const { longPressProps } = (0, import_interactions.useLongPress)({
    isDisabled: props.longPressDisabled,
    accessibilityDescription: "Long press to open context menu",
    onLongPressStart: (e) => {
      onClose();
    },
    onLongPress: (e) => {
      if (e.pointerType === "mouse")
        return;
      if (e.type === "longpress") {
        onOpen(e);
        openAndFocusFirstItem();
      }
    }
  });
  const anchorRef = React8.useRef({
    getBoundingClientRect: generateClientRect(anchor.x, anchor.y)
  });
  React8.useEffect(() => {
    popper.referenceRef(anchorRef.current);
  }, []);
  React8.useEffect(() => {
    anchorRef.current.getBoundingClientRect = generateClientRect(
      anchor.x,
      anchor.y
    );
    menu.popper.update();
  }, [anchor]);
  const onPointerDown = (event) => {
    var _a;
    if (event.pointerType !== "mouse") {
      (_a = longPressProps.onPointerDown) == null ? void 0 : _a.call(longPressProps, event);
    }
  };
  const onMouseDown = (event) => {
    var _a;
    if (isTouchDevice()) {
      (_a = longPressProps.onMouseDown) == null ? void 0 : _a.call(longPressProps, event);
    }
  };
  return {
    triggerProps: {
      ...longPressProps,
      onPointerDown,
      onMouseDown,
      onContextMenu: (0, import_utils8.callAllHandlers)((event) => {
        event.preventDefault();
        onOpen(event);
        openAndFocusFirstItem();
      }, props.onContextMenu),
      ref: (0, import_react_utils5.mergeRefs)(triggerRef, ref)
    }
  };
};
var ContextMenuTrigger = (0, import_react21.forwardRef)(
  (props, ref) => {
    const { children, longPressDisabled, ...rest } = props;
    const { triggerProps } = useContextMenuTrigger(props, ref);
    return /* @__PURE__ */ (0, import_jsx_runtime18.jsx)(
      import_react21.chakra.span,
      {
        ...rest,
        sx: { WebkitTouchCallout: "none" },
        ...triggerProps,
        children
      }
    );
  }
);
ContextMenuTrigger.displayName = "ContextMenuTrigger";
var ContextMenuList = (props) => {
  const { children, ...rest } = props;
  const { menuRef } = useContextMenuContext();
  return /* @__PURE__ */ (0, import_jsx_runtime18.jsx)(import_react21.MenuList, { ref: menuRef, ...rest, children });
};
ContextMenuList.displayName = "ContextMenuList";
var ContextMenuItem = import_react21.MenuItem;

// src/persona/persona.tsx
var import_react22 = require("@chakra-ui/react");
var import_utils9 = require("@chakra-ui/utils");
var import_jsx_runtime19 = require("react/jsx-runtime");
var [StylesProvider6, useStyles6] = (0, import_react22.createStylesContext)("SuiPersona");
var Presence = {
  online: {
    label: "Online",
    color: "presence.online"
  },
  offline: {
    label: "Offline",
    color: "presence.offline"
  },
  busy: {
    label: "Busy",
    color: "presence.busy"
  },
  dnd: {
    label: "Do-not-disturb",
    color: "presence.dnd"
  },
  away: {
    label: "Away",
    color: "presence.away"
  }
};
var defaultPresenceTokens = {
  online: "green.500",
  offline: "gray.400",
  busy: "orange.500",
  dnd: "red.500",
  away: "gray.400"
};
var Persona = (props) => {
  const {
    name,
    presence,
    presenceIcon,
    isOutOfOffice,
    label,
    secondaryLabel,
    tertiaryLabel,
    size,
    hideDetails,
    children,
    /** Avatar props */
    getInitials,
    icon,
    iconLabel,
    ignoreFallback,
    loading,
    onError,
    src,
    srcSet,
    ...rest
  } = props;
  const isComposed = !!(name || label || src || icon);
  return /* @__PURE__ */ (0, import_jsx_runtime19.jsx)(PersonaContainer, { size, ...rest, children: isComposed ? /* @__PURE__ */ (0, import_jsx_runtime19.jsxs)(import_jsx_runtime19.Fragment, { children: [
    /* @__PURE__ */ (0, import_jsx_runtime19.jsx)(
      PersonaAvatar,
      {
        name,
        presence,
        presenceIcon,
        isOutOfOffice,
        size,
        getInitials,
        icon,
        iconLabel,
        ignoreFallback,
        loading,
        onError,
        src,
        srcSet
      }
    ),
    !hideDetails && /* @__PURE__ */ (0, import_jsx_runtime19.jsxs)(PersonaDetails, { children: [
      /* @__PURE__ */ (0, import_jsx_runtime19.jsx)(PersonaLabel, { children: label || name }),
      secondaryLabel && /* @__PURE__ */ (0, import_jsx_runtime19.jsx)(PersonaSecondaryLabel, { children: secondaryLabel }),
      tertiaryLabel && /* @__PURE__ */ (0, import_jsx_runtime19.jsx)(PersonaTertiaryLabel, { children: tertiaryLabel }),
      children
    ] })
  ] }) : children });
};
Persona.displayName = "Persona";
var PersonaContainer = (0, import_react22.forwardRef)(
  (props, ref) => {
    const { children, ...rest } = props;
    const styles = (0, import_react22.useMultiStyleConfig)("SuiPersona", props);
    const containerProps = (0, import_react22.omitThemingProps)(rest);
    const baseStyle = {
      display: "flex",
      flexDirection: "row",
      alignItems: "center"
    };
    const containerStyles = {
      ...baseStyle,
      ...styles.container
    };
    return /* @__PURE__ */ (0, import_jsx_runtime19.jsx)(StylesProvider6, { value: styles, children: /* @__PURE__ */ (0, import_jsx_runtime19.jsx)(
      import_react22.chakra.div,
      {
        ref,
        __css: containerStyles,
        ...containerProps,
        className: (0, import_utils9.cx)("sui-persona", props.className),
        children
      }
    ) });
  }
);
PersonaContainer.displayName = "PersonaContainer";
var PersonaAvatar = (0, import_react22.forwardRef)(
  (props, ref) => {
    var _a, _b, _c, _d, _e;
    const {
      name,
      presence,
      presenceLabel,
      presenceIcon,
      isOutOfOffice,
      badgeSize = "1em",
      size,
      getInitials,
      icon,
      iconLabel,
      ignoreFallback,
      loading,
      onError,
      src,
      srcSet,
      ...rest
    } = props;
    const badgeProps = {};
    let badge;
    const theme = (0, import_react22.useTheme)();
    const colors = ((_a = theme.colors) == null ? void 0 : _a.presence) || defaultPresenceTokens;
    const semantic = !!((_c = (_b = theme.semanticTokens) == null ? void 0 : _b.colors) == null ? void 0 : _c["presence.online"]);
    if (presence) {
      const label = presenceLabel || ((_d = Presence[presence]) == null ? void 0 : _d.label);
      const color = semantic ? ((_e = Presence[presence]) == null ? void 0 : _e.color) || `presence.${presence}` : colors[presence];
      if (isOutOfOffice) {
        badgeProps.sx = {
          _before: {
            content: '""',
            width: "100%",
            height: "100%",
            position: "absolute",
            top: 0,
            left: 0,
            border: "0.2em solid",
            borderColor: color,
            borderRadius: "50%",
            boxSizing: "border-box"
          }
        };
        badgeProps.borderWidth = "0.15em";
        badgeProps.bg = (0, import_react22.useColorModeValue)("white", "gray.800");
      } else {
        badgeProps.bg = color;
      }
      badge = /* @__PURE__ */ (0, import_jsx_runtime19.jsx)(import_react22.AvatarBadge, { boxSize: badgeSize, ...badgeProps, children: presenceIcon });
      if (label) {
        badge = /* @__PURE__ */ (0, import_jsx_runtime19.jsx)(import_react22.Tooltip, { label, children: badge });
      }
    }
    return /* @__PURE__ */ (0, import_jsx_runtime19.jsx)(
      import_react22.Avatar,
      {
        ref,
        name,
        size,
        getInitials,
        icon,
        iconLabel,
        ignoreFallback,
        loading,
        onError,
        src,
        srcSet,
        ...rest,
        children: badge
      }
    );
  }
);
PersonaAvatar.displayName = "PersonaAvatar";
var PersonaDetails = (0, import_react22.forwardRef)((props, ref) => {
  const { children, className, ...rest } = props;
  const styles = useStyles6();
  const baseStyle = {
    display: "flex",
    flexDirection: "column"
  };
  const detailsStyles = {
    ...baseStyle,
    ...styles.details
  };
  return /* @__PURE__ */ (0, import_jsx_runtime19.jsx)(
    import_react22.chakra.div,
    {
      ref,
      ...rest,
      __css: detailsStyles,
      className: (0, import_utils9.cx)("sui-persona__details", className),
      children
    }
  );
});
PersonaDetails.displayName = "PersonaDetails";
var PersonaLabel = (0, import_react22.forwardRef)(
  (props, ref) => {
    const styles = useStyles6();
    return /* @__PURE__ */ (0, import_jsx_runtime19.jsx)(
      import_react22.chakra.span,
      {
        ref,
        ...props,
        __css: styles.label,
        className: (0, import_utils9.cx)("sui-persona__label", props.className)
      }
    );
  }
);
PersonaLabel.displayName = "PersonaLabel";
var PersonaSecondaryLabel = (0, import_react22.forwardRef)((props, ref) => {
  const styles = useStyles6();
  return /* @__PURE__ */ (0, import_jsx_runtime19.jsx)(
    import_react22.chakra.span,
    {
      ref,
      ...props,
      __css: styles.secondaryLabel,
      className: (0, import_utils9.cx)("sui-persona__secondary-label", props.className)
    }
  );
});
PersonaSecondaryLabel.displayName = "PersonaSecondaryLabel";
var PersonaTertiaryLabel = (0, import_react22.forwardRef)(
  (props, ref) => {
    const styles = useStyles6();
    return /* @__PURE__ */ (0, import_jsx_runtime19.jsx)(
      import_react22.chakra.span,
      {
        ref,
        ...props,
        __css: styles.tertiaryLabel,
        className: (0, import_utils9.cx)("sui-persona__tertiary-label", props.className)
      }
    );
  }
);
PersonaTertiaryLabel.displayName = "PersonaTertiaryLabel";

// src/property/property.tsx
var React9 = __toESM(require("react"));
var import_react23 = require("@chakra-ui/react");
var import_utils10 = require("@chakra-ui/utils");
var import_jsx_runtime20 = require("react/jsx-runtime");
var [StylesProvider7, useStyles7] = (0, import_react23.createStylesContext)("SuiProperty");
var Property = (0, import_react23.forwardRef)((props, ref) => {
  const styles = (0, import_react23.useMultiStyleConfig)("SuiProperty", props);
  const { children, label, value, labelWidth, spacing, ...rest } = (0, import_react23.omitThemingProps)(props);
  const propertyStyles = {
    minW: 0,
    display: "flex",
    flexDirection: "row",
    alignItems: "center",
    ...styles.property
  };
  return /* @__PURE__ */ (0, import_jsx_runtime20.jsx)(StylesProvider7, { value: styles, children: /* @__PURE__ */ (0, import_jsx_runtime20.jsxs)(
    import_react23.chakra.dl,
    {
      ref,
      __css: propertyStyles,
      ...rest,
      className: (0, import_utils10.cx)("sui-property", props.className),
      children: [
        label && /* @__PURE__ */ (0, import_jsx_runtime20.jsx)(
          PropertyLabel,
          {
            width: labelWidth,
            minWidth: labelWidth,
            marginEnd: spacing,
            children: label
          }
        ),
        value && /* @__PURE__ */ (0, import_jsx_runtime20.jsx)(PropertyValue, { children: value }),
        children
      ]
    }
  ) });
});
Property.displayName = "Property";
var PropertyLabel = (0, import_react23.forwardRef)((props, ref) => {
  const styles = useStyles7();
  const { children, noOfLines = 1, width, minWidth, ...rest } = props;
  const labelStyles = {
    display: "flex",
    flexDirection: "row",
    ...styles.label
  };
  if (width) {
    labelStyles.minWidth = minWidth || "auto";
    labelStyles.width = width;
  }
  return /* @__PURE__ */ (0, import_jsx_runtime20.jsx)(
    import_react23.chakra.dt,
    {
      ref,
      __css: labelStyles,
      ...rest,
      className: (0, import_utils10.cx)("sui-property__label", props.className),
      children: /* @__PURE__ */ (0, import_jsx_runtime20.jsx)(import_react23.chakra.span, { flex: "1", noOfLines, children })
    }
  );
});
PropertyLabel.displayName = "PropertyLabel";
var PropertyValue = (0, import_react23.forwardRef)(
  (props, ref) => {
    const styles = useStyles7();
    const { children, ...rest } = props;
    const valueStyles = {
      display: "flex",
      flexDirection: "row",
      alignItems: "center",
      flex: 1,
      ...styles.value
    };
    return /* @__PURE__ */ (0, import_jsx_runtime20.jsx)(
      import_react23.chakra.dd,
      {
        ref,
        __css: valueStyles,
        ...rest,
        className: (0, import_utils10.cx)("sui-property__value", props.className),
        children
      }
    );
  }
);
PropertyValue.displayName = "PropertyValue";
var PropertyList = (props) => {
  const { children, ...rest } = props;
  return /* @__PURE__ */ (0, import_jsx_runtime20.jsx)(import_react23.chakra.dl, { ...rest, className: (0, import_utils10.cx)("sui-property-list", props.className), children: React9.Children.map(
    children,
    (child) => React9.isValidElement(child) ? React9.cloneElement(child, {
      as: "div"
    }) : child
  ) });
};
PropertyList.displayName = "PropertyList";

// src/navbar/use-navbar.ts
var import_utils11 = require("@chakra-ui/utils");
var import_react28 = require("react");
var import_utils12 = require("@react-aria/utils");

// ../saas-ui-hooks/dist/index.mjs
var React10 = __toESM(require("react"), 1);
var import_react24 = __toESM(require("react"), 1);
var import_react25 = require("react");
var import_react26 = require("react");
var import_react27 = require("react");
var isBrowser2 = typeof window !== "undefined";
function getScrollPosition(element) {
  if (!isBrowser2)
    return { x: 0, y: 0 };
  if (!element) {
    return { x: window.scrollX, y: window.scrollY };
  }
  return { x: element.scrollLeft, y: element.scrollTop };
}
var useScrollPosition = (props) => {
  const { elementRef, delay = 30, callback, isEnabled } = props;
  const position = (0, import_react27.useRef)(
    isEnabled ? getScrollPosition(elementRef == null ? void 0 : elementRef.current) : { x: 0, y: 0 }
  );
  let throttleTimeout = null;
  const handler = () => {
    const currPos = getScrollPosition(elementRef == null ? void 0 : elementRef.current);
    if (typeof callback === "function") {
      callback({ prevPos: position.current, currPos });
    }
    position.current = currPos;
    throttleTimeout = null;
  };
  (0, import_react27.useEffect)(() => {
    if (!isEnabled)
      return;
    const handleScroll = () => {
      if (delay) {
        if (throttleTimeout === null) {
          throttleTimeout = setTimeout(handler, delay);
        }
      } else {
        handler();
      }
    };
    const target = (elementRef == null ? void 0 : elementRef.current) || window;
    target.addEventListener("scroll", handleScroll);
    return () => target.removeEventListener("scroll", handleScroll);
  }, [elementRef == null ? void 0 : elementRef.current, delay, isEnabled]);
  return position.current;
};

// src/navbar/use-navbar.ts
function useNavbar(props) {
  const {
    ref,
    parentRef,
    height = "3.5rem",
    shouldHideOnScroll = false,
    disableScrollHandler = false,
    onScrollPositionChange,
    motionProps,
    ...containerProps
  } = props;
  const containerRef = (0, import_react28.useRef)(null);
  (0, import_react28.useImperativeHandle)(ref, () => containerRef.current);
  const prevWidth = (0, import_react28.useRef)(0);
  const navHeight = (0, import_react28.useRef)(0);
  const [isHidden2, setIsHidden] = (0, import_react28.useState)(false);
  const updateWidth = () => {
    if (containerRef.current) {
      const width = containerRef.current.offsetWidth;
      if (width !== prevWidth.current) {
        prevWidth.current = width;
      }
    }
  };
  (0, import_utils12.useResizeObserver)({
    ref: containerRef,
    onResize: () => {
      var _a;
      const currentWidth = (_a = containerRef.current) == null ? void 0 : _a.offsetWidth;
      if (currentWidth !== prevWidth.current) {
        updateWidth();
      }
    }
  });
  (0, import_react28.useEffect)(() => {
    var _a;
    updateWidth();
    navHeight.current = ((_a = containerRef.current) == null ? void 0 : _a.offsetHeight) || 0;
  }, []);
  useScrollPosition({
    elementRef: parentRef,
    isEnabled: shouldHideOnScroll || !disableScrollHandler,
    callback: ({ prevPos, currPos }) => {
      onScrollPositionChange == null ? void 0 : onScrollPositionChange(currPos.y);
      if (shouldHideOnScroll) {
        setIsHidden((prev) => {
          const next = currPos.y > prevPos.y && currPos.y > navHeight.current;
          return next !== prev ? next : prev;
        });
      }
    }
  });
  const getContainerProps = (props2 = {}) => ({
    ...containerProps,
    ...motionProps,
    "data-hidden": (0, import_utils11.dataAttr)(isHidden2),
    ref: containerRef,
    style: {
      "--navbar-height": height,
      ...containerProps.style,
      ...props2 == null ? void 0 : props2.style
    }
  });
  return {
    containerRef,
    height,
    isHidden: isHidden2,
    shouldHideOnScroll,
    motionProps,
    getContainerProps
  };
}

// ../../node_modules/@chakra-ui/react-context/dist/index.mjs
var import_react29 = require("react");
function getErrorMessage(hook, provider) {
  return `${hook} returned \`undefined\`. Seems you forgot to wrap component within ${provider}`;
}
function createContext7(options = {}) {
  const {
    name,
    strict = true,
    hookName = "useContext",
    providerName = "Provider",
    errorMessage,
    defaultValue
  } = options;
  const Context = (0, import_react29.createContext)(defaultValue);
  Context.displayName = name;
  function useContext2() {
    var _a;
    const context = (0, import_react29.useContext)(Context);
    if (!context && strict) {
      const error2 = new Error(
        errorMessage != null ? errorMessage : getErrorMessage(hookName, providerName)
      );
      error2.name = "ContextError";
      (_a = Error.captureStackTrace) == null ? void 0 : _a.call(Error, error2, useContext2);
      throw error2;
    }
    return context;
  }
  return [Context.Provider, useContext2, Context];
}

// src/navbar/navbar-context.ts
var [NavbarProvider, useNavbarContext] = createContext7({
  name: "NavbarContext",
  strict: true,
  errorMessage: "useNavbarContext: `context` is undefined. Seems you forgot to wrap component within <Navbar />"
});
var [NavBarStylesProvider, useNavBarStyles] = createContext7({
  name: `NavBarStylesContext`,
  hookName: `useNavItemStyles`,
  providerName: "<NavBar />"
});

// src/navbar/navbar.tsx
var import_react30 = require("@chakra-ui/react");
var import_framer_motion6 = require("framer-motion");
var import_react_utils6 = require("@saas-ui/react-utils");
var import_jsx_runtime21 = require("react/jsx-runtime");
var MotionBox = (0, import_react30.chakra)(import_framer_motion6.motion.nav);
var Navbar = (0, import_react30.forwardRef)((props, ref) => {
  const { children, ...otherProps } = props;
  const context = useNavbar({ ...otherProps, ref });
  const styles = (0, import_react30.useMultiStyleConfig)("SuiNavbar", props);
  const content = /* @__PURE__ */ (0, import_jsx_runtime21.jsx)(import_react30.chakra.header, { __css: styles.inner, className: "sui-navbar__inner", children });
  const containerStyles = {
    top: props.position === "sticky" ? "0" : void 0,
    insetX: props.position === "sticky" ? "0" : void 0,
    ...styles.container
  };
  return /* @__PURE__ */ (0, import_jsx_runtime21.jsx)(NavBarStylesProvider, { value: styles, children: /* @__PURE__ */ (0, import_jsx_runtime21.jsx)(NavbarProvider, { value: context, children: /* @__PURE__ */ (0, import_jsx_runtime21.jsx)(
    MotionBox,
    {
      __css: containerStyles,
      animate: context.isHidden ? "hidden" : "visible",
      initial: false,
      variants: {
        hidden: { y: "-100%" },
        visible: { y: 0, transition: { ease: "easeInOut" } }
      },
      className: (0, import_react_utils6.cx)("sui-navbar", props.className),
      ...context.getContainerProps(),
      children: content
    }
  ) }) });
});
Navbar.displayName = "Navbar";

// src/navbar/navbar-brand.tsx
var import_react31 = require("@chakra-ui/react");
var import_utils13 = require("@chakra-ui/utils");
var import_jsx_runtime22 = require("react/jsx-runtime");
var NavbarBrand = (0, import_react31.forwardRef)((props, ref) => {
  const { className, children, ...rest } = props;
  const styles = useNavBarStyles();
  return /* @__PURE__ */ (0, import_jsx_runtime22.jsx)(
    import_react31.chakra.div,
    {
      ref,
      __css: styles.brand,
      className: (0, import_utils13.cx)("sui-navbar__brand"),
      ...rest,
      children
    }
  );
});
NavbarBrand.displayName = "NavbarBrand";

// src/navbar/navbar-content.tsx
var import_react32 = require("@chakra-ui/react");
var import_utils14 = require("@chakra-ui/utils");
var import_jsx_runtime23 = require("react/jsx-runtime");
var NavbarContent = (0, import_react32.forwardRef)(
  (props, ref) => {
    const { className, children, spacing = 0, ...rest } = props;
    const styles = useNavBarStyles();
    const contentStyles = {
      ...styles.content,
      "& > *:not(style) ~ *:not(style)": { marginStart: spacing }
    };
    return /* @__PURE__ */ (0, import_jsx_runtime23.jsx)(
      import_react32.chakra.ul,
      {
        ref,
        __css: contentStyles,
        className: (0, import_utils14.cx)("sui-navbar__content", className),
        ...rest,
        children
      }
    );
  }
);
NavbarContent.displayName = "NavbarContent";

// src/navbar/navbar-item.tsx
var import_react33 = require("@chakra-ui/react");
var import_utils15 = require("@chakra-ui/utils");
var import_jsx_runtime24 = require("react/jsx-runtime");
var NavbarItem = (0, import_react33.forwardRef)((props, ref) => {
  const { className, children, isActive, ...rest } = props;
  const styles = useNavBarStyles();
  return /* @__PURE__ */ (0, import_jsx_runtime24.jsx)(
    import_react33.chakra.li,
    {
      ref,
      __css: styles.item,
      className: (0, import_utils15.cx)("sui-navbar__item", className),
      "data-active": (0, import_utils15.dataAttr)(isActive),
      ...rest,
      children
    }
  );
});
NavbarItem.displayName = "NavbarItem";

// src/navbar/navbar-link.tsx
var import_react34 = require("@chakra-ui/react");
var import_utils16 = require("@chakra-ui/utils");
var import_jsx_runtime25 = require("react/jsx-runtime");
var NavbarLink = (0, import_react34.forwardRef)((props, ref) => {
  const { className, children, isActive, ...rest } = props;
  const Link3 = useLink();
  const styles = useNavBarStyles();
  return /* @__PURE__ */ (0, import_jsx_runtime25.jsx)(
    import_react34.chakra.a,
    {
      as: Link3,
      ref,
      __css: styles.link,
      "data-active": (0, import_utils16.dataAttr)(isActive),
      className: (0, import_utils16.cx)("sui-navbar__link", className),
      ...rest,
      children
    }
  );
});
NavbarLink.displayName = "NavbarLink";

// src/sidebar/sidebar.tsx
var React12 = __toESM(require("react"));
var import_react36 = require("@chakra-ui/react");
var import_utils18 = require("@chakra-ui/utils");

// ../../node_modules/@chakra-ui/icon/dist/chunk-2GBDXOMA.mjs
var import_system2 = require("@chakra-ui/system");
var import_jsx_runtime26 = require("react/jsx-runtime");
var fallbackIcon = {
  path: /* @__PURE__ */ (0, import_jsx_runtime26.jsxs)("g", { stroke: "currentColor", strokeWidth: "1.5", children: [
    /* @__PURE__ */ (0, import_jsx_runtime26.jsx)(
      "path",
      {
        strokeLinecap: "round",
        fill: "none",
        d: "M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
      }
    ),
    /* @__PURE__ */ (0, import_jsx_runtime26.jsx)(
      "path",
      {
        fill: "currentColor",
        strokeLinecap: "round",
        d: "M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
      }
    ),
    /* @__PURE__ */ (0, import_jsx_runtime26.jsx)("circle", { fill: "none", strokeMiterlimit: "10", cx: "12", cy: "12", r: "11.25" })
  ] }),
  viewBox: "0 0 24 24"
};
var Icon6 = (0, import_system2.forwardRef)((props, ref) => {
  const {
    as: element,
    viewBox,
    color = "currentColor",
    focusable = false,
    children,
    className,
    __css,
    ...rest
  } = props;
  const _className = cx4("chakra-icon", className);
  const customStyles = (0, import_system2.useStyleConfig)("Icon", props);
  const styles = {
    w: "1em",
    h: "1em",
    display: "inline-block",
    lineHeight: "1em",
    flexShrink: 0,
    color,
    ...__css,
    ...customStyles
  };
  const shared = {
    ref,
    focusable,
    className: _className,
    __css: styles
  };
  const _viewBox = viewBox != null ? viewBox : fallbackIcon.viewBox;
  if (element && typeof element !== "string") {
    return /* @__PURE__ */ (0, import_jsx_runtime26.jsx)(import_system2.chakra.svg, { as: element, ...shared, ...rest });
  }
  const _path = children != null ? children : fallbackIcon.path;
  return /* @__PURE__ */ (0, import_jsx_runtime26.jsx)(import_system2.chakra.svg, { verticalAlign: "middle", viewBox: _viewBox, ...shared, ...rest, children: _path });
});
Icon6.displayName = "Icon";

// ../../node_modules/@chakra-ui/icon/dist/chunk-DEQZ7DVA.mjs
var import_system3 = require("@chakra-ui/system");
var import_react35 = require("react");
var import_jsx_runtime27 = require("react/jsx-runtime");
function createIcon(options) {
  const {
    viewBox = "0 0 24 24",
    d: pathDefinition,
    displayName,
    defaultProps = {}
  } = options;
  const path = import_react35.Children.toArray(options.path);
  const Comp = (0, import_system3.forwardRef)((props, ref) => /* @__PURE__ */ (0, import_jsx_runtime27.jsx)(Icon6, { ref, viewBox, ...defaultProps, ...props, children: path.length ? path : /* @__PURE__ */ (0, import_jsx_runtime27.jsx)("path", { fill: "currentColor", d: pathDefinition }) }));
  Comp.displayName = displayName;
  return Comp;
}

// src/icons/create-icon.tsx
var createIcon2 = (props) => {
  return createIcon({
    viewBox: "0 0 24 24",
    defaultProps: {
      fill: "none",
      stroke: "currentColor",
      strokeWidth: "2",
      strokeLinecap: "round",
      strokeLinejoin: "round"
    },
    ...props
  });
};

// src/icons/icons.tsx
var import_jsx_runtime28 = require("react/jsx-runtime");
var ChevronUpIcon = createIcon2({
  displayName: "ChevronUpIcon",
  path: /* @__PURE__ */ (0, import_jsx_runtime28.jsx)("polyline", { points: "18 15 12 9 6 15" })
});
var ChevronDownIcon = createIcon2({
  displayName: "ChevronDownIcon",
  path: /* @__PURE__ */ (0, import_jsx_runtime28.jsx)("polyline", { points: "6 9 12 15 18 9" })
});
var ChevronLeftIcon = createIcon2({
  displayName: "ChevronLeftIcon",
  path: /* @__PURE__ */ (0, import_jsx_runtime28.jsx)("polyline", { points: "15 18 9 12 15 6" })
});
var ChevronRightIcon = createIcon2({
  displayName: "ChevronRightIcon",
  path: /* @__PURE__ */ (0, import_jsx_runtime28.jsx)("polyline", { points: "9 18 15 12 9 6" })
});
var HamburgerIcon = createIcon2({
  displayName: "ChevronDownIcon",
  path: /* @__PURE__ */ (0, import_jsx_runtime28.jsxs)("g", { fill: "none", children: [
    /* @__PURE__ */ (0, import_jsx_runtime28.jsx)("line", { x1: "3", y1: "12", x2: "21", y2: "12" }),
    /* @__PURE__ */ (0, import_jsx_runtime28.jsx)("line", { x1: "3", y1: "6", x2: "21", y2: "6" }),
    /* @__PURE__ */ (0, import_jsx_runtime28.jsx)("line", { x1: "3", y1: "18", x2: "21", y2: "18" })
  ] })
});
var CloseIcon = createIcon2({
  displayName: "CloseIcon",
  path: /* @__PURE__ */ (0, import_jsx_runtime28.jsxs)("g", { children: [
    /* @__PURE__ */ (0, import_jsx_runtime28.jsx)("line", { x1: "18", y1: "6", x2: "6", y2: "18" }),
    /* @__PURE__ */ (0, import_jsx_runtime28.jsx)("line", { x1: "6", y1: "6", x2: "18", y2: "18" })
  ] })
});
var FilterIcon = createIcon2({
  displayName: "FilterIcon",
  path: /* @__PURE__ */ (0, import_jsx_runtime28.jsx)("polygon", { points: "22 3 2 3 10 12.46 10 19 14 21 14 12.46 22 3" })
});
var CalendarIcon = createIcon2({
  displayName: "CalendarIcon",
  path: /* @__PURE__ */ (0, import_jsx_runtime28.jsxs)("g", { children: [
    /* @__PURE__ */ (0, import_jsx_runtime28.jsx)("rect", { x: "3", y: "4", width: "18", height: "18", rx: "2", ry: "2" }),
    /* @__PURE__ */ (0, import_jsx_runtime28.jsx)("line", { x1: "16", y1: "2", x2: "16", y2: "6" }),
    /* @__PURE__ */ (0, import_jsx_runtime28.jsx)("line", { x1: "8", y1: "2", x2: "8", y2: "6" }),
    /* @__PURE__ */ (0, import_jsx_runtime28.jsx)("line", { x1: "3", y1: "10", x2: "21", y2: "10" })
  ] })
});
var PlusIcon = createIcon2({
  displayName: "PlusIcon",
  path: /* @__PURE__ */ (0, import_jsx_runtime28.jsxs)("g", { children: [
    /* @__PURE__ */ (0, import_jsx_runtime28.jsx)("line", { x1: "12", y1: "5", x2: "12", y2: "19" }),
    /* @__PURE__ */ (0, import_jsx_runtime28.jsx)("line", { x1: "5", y1: "12", x2: "19", y2: "12" })
  ] })
});
var MinusIcon = createIcon2({
  displayName: "MinusIcon",
  path: /* @__PURE__ */ (0, import_jsx_runtime28.jsx)("g", { children: /* @__PURE__ */ (0, import_jsx_runtime28.jsx)("line", { x1: "5", y1: "12", x2: "19", y2: "12" }) })
});
var ViewOffIcon = createIcon2({
  displayName: "ViewOffIcon",
  path: /* @__PURE__ */ (0, import_jsx_runtime28.jsxs)("g", { children: [
    /* @__PURE__ */ (0, import_jsx_runtime28.jsx)("path", { d: "M17.94 17.94A10.07 10.07 0 0 1 12 20c-7 0-11-8-11-8a18.45 18.45 0 0 1 5.06-5.94M9.9 4.24A9.12 9.12 0 0 1 12 4c7 0 11 8 11 8a18.5 18.5 0 0 1-2.16 3.19m-6.72-1.07a3 3 0 1 1-4.24-4.24" }),
    /* @__PURE__ */ (0, import_jsx_runtime28.jsx)("line", { x1: "1", y1: "1", x2: "23", y2: "23" })
  ] })
});
var ViewIcon = createIcon2({
  displayName: "ViewOffIcon",
  path: /* @__PURE__ */ (0, import_jsx_runtime28.jsxs)("g", { children: [
    /* @__PURE__ */ (0, import_jsx_runtime28.jsx)("path", { d: "M1 12s4-8 11-8 11 8 11 8-4 8-11 8-11-8-11-8z" }),
    /* @__PURE__ */ (0, import_jsx_runtime28.jsx)("circle", { cx: "12", cy: "12", r: "3" })
  ] })
});
var SearchIcon = createIcon2({
  displayName: "SearchIcon",
  path: /* @__PURE__ */ (0, import_jsx_runtime28.jsxs)("g", { children: [
    /* @__PURE__ */ (0, import_jsx_runtime28.jsx)("circle", { cx: "11", cy: "11", r: "8" }),
    /* @__PURE__ */ (0, import_jsx_runtime28.jsx)("line", { x1: "21", y1: "21", x2: "16.65", y2: "16.65" })
  ] })
});
var CheckIcon2 = createIcon2({
  displayName: "CheckIcon",
  path: /* @__PURE__ */ (0, import_jsx_runtime28.jsx)("g", { children: /* @__PURE__ */ (0, import_jsx_runtime28.jsx)("polyline", { points: "20 6 9 17 4 12" }) })
});

// src/sidebar/sidebar.tsx
var import_framer_motion7 = require("framer-motion");
var import_react_utils8 = require("@saas-ui/react-utils");

// src/sidebar/use-sidebar.ts
var import_react_utils7 = require("@chakra-ui/react-utils");
var import_utils17 = require("@chakra-ui/utils");
var [SidebarProvider, useSidebarContext] = (0, import_react_utils7.createContext)({
  name: "SidebarContext",
  strict: false
});
var useSidebarToggleButton = () => {
  const appShellContext = useAppShellContext();
  const context = useSidebarContext();
  const getButtonProps = (props) => {
    return {
      onClick: (0, import_utils17.callAllHandlers)(
        (context == null ? void 0 : context.onToggle) || (appShellContext == null ? void 0 : appShellContext.toggleSidebar),
        props == null ? void 0 : props.onClick
      )
    };
  };
  return {
    isOpen: (context == null ? void 0 : context.isOpen) || (appShellContext == null ? void 0 : appShellContext.isSidebarOpen),
    isMobile: (context == null ? void 0 : context.isMobile) || (appShellContext == null ? void 0 : appShellContext.isMobile),
    getButtonProps
  };
};

// src/sidebar/sidebar-context.tsx
var [SidebarStylesProvider, useSidebarStyles] = createContext7({
  name: `SidebarStylesContext`,
  hookName: `useSidebarStyles`,
  providerName: "<Sidebar />"
});

// src/sidebar/sidebar.tsx
var import_jsx_runtime29 = require("react/jsx-runtime");
var MotionBox2 = (0, import_react36.chakra)(import_framer_motion7.motion.nav);
var motionPresets = {
  slideInOut: {
    enter: {
      left: 0,
      transition: { type: "spring", duration: 0.6, bounce: 0.15 }
    },
    exit: {
      left: "-100%"
    }
  },
  none: {}
};
var Sidebar = (0, import_react36.forwardRef)((props, ref) => {
  var _a, _b, _c;
  const styles = (0, import_react36.useMultiStyleConfig)("SuiSidebar", props);
  const theme = (0, import_react36.useTheme)();
  const defaultProps = (_a = theme.components["SuiSidebar"]) == null ? void 0 : _a.defaultProps;
  const variant = (0, import_react_utils8.useResponsiveValue)((_b = props.variant) != null ? _b : defaultProps == null ? void 0 : defaultProps.variant, {
    fallback: "base"
  });
  const size = (0, import_react_utils8.useResponsiveValue)((_c = props.size) != null ? _c : defaultProps == null ? void 0 : defaultProps.size, {
    fallback: "base"
  });
  const isCondensed = variant === "compact";
  const {
    spacing = 4,
    children,
    toggleBreakpoint = "lg",
    className,
    motionPreset = "slideInOut",
    isOpen: isOpenProp,
    onOpen: onOpenProp,
    onClose: onCloseProp,
    ...containerProps
  } = (0, import_react36.omitThemingProps)(props);
  const appShell = useAppShellContext();
  const breakpoints2 = getBreakpoints(toggleBreakpoint);
  const isMobile = (0, import_react36.useBreakpointValue)(breakpoints2, {
    fallback: void 0
  });
  const isMobileInitial = (0, import_react36.useBreakpointValue)(breakpoints2);
  const isInitial = typeof isMobile === "undefined";
  const isControlled = typeof isOpenProp !== "undefined";
  const isCollapsible = (isMobile || isControlled) && !isCondensed;
  const disclosure = (0, import_react36.useDisclosure)({
    isOpen: isOpenProp || (appShell == null ? void 0 : appShell.isSidebarOpen),
    onOpen: onOpenProp || (appShell == null ? void 0 : appShell.openSidebar),
    onClose: onCloseProp || (appShell == null ? void 0 : appShell.closeSidebar)
  });
  const { isOpen, onClose, onOpen } = disclosure;
  React12.useEffect(() => {
    if (isInitial && isMobileInitial || isCondensed || isControlled) {
      return;
    }
    isMobileInitial ? onClose() : onOpen();
  }, [isInitial, isCondensed, isMobileInitial]);
  const containerStyles = {
    "& > *:not(style) ~ *:not(style, .sui-resize-handle, .sui-sidebar__toggle-button + *)": {
      marginTop: spacing
    },
    display: "flex",
    flexDirection: "column",
    ...isMobile && isCollapsible ? {
      position: "absolute",
      zIndex: "modal",
      top: 0,
      left: { base: "-100%", lg: "0" },
      bottom: 0
    } : {
      position: "relative"
    }
  };
  const context = {
    ...disclosure,
    breakpoints: breakpoints2,
    isMobile,
    variant,
    size
  };
  const variants3 = motionPresets[isCondensed ? "none" : motionPreset || "none"];
  return /* @__PURE__ */ (0, import_jsx_runtime29.jsx)(SidebarProvider, { value: context, children: /* @__PURE__ */ (0, import_jsx_runtime29.jsx)(SidebarStylesProvider, { value: styles, children: /* @__PURE__ */ (0, import_jsx_runtime29.jsx)(
    MotionBox2,
    {
      ref,
      initial: false,
      animate: !isInitial && (!isCollapsible || isOpen ? "enter" : "exit"),
      variants: variants3,
      __css: {
        ...containerStyles,
        ...styles.container
      },
      ...containerProps,
      id: disclosure.getDisclosureProps().id,
      className: (0, import_utils18.cx)("sui-sidebar", className),
      "data-compact": (0, import_utils18.dataAttr)(isCondensed),
      "data-collapsible": (0, import_utils18.dataAttr)(isMobile && isCollapsible),
      children
    }
  ) }) });
});
Sidebar.displayName = "Sidebar";
Sidebar.id = "Sidebar";
var SidebarToggleButton = (props) => {
  var _a, _b;
  const { sx, pos, position, variant = "ghost", icon, isRound, ...rest } = props;
  const { isOpen, isMobile, getButtonProps } = useSidebarToggleButton();
  const styles = (0, import_react36.useStyleConfig)("SuiSidebarToggleButton", props);
  const btnStyles = (0, import_react36.useStyleConfig)("Button", {
    ...props,
    variant
  });
  const p = (_b = (_a = pos != null ? pos : position) != null ? _a : sx == null ? void 0 : sx.pos) != null ? _b : sx == null ? void 0 : sx.position;
  const buttonStyles = {
    display: "inline-flex",
    appearance: "none",
    alignItems: "center",
    justifyContent: "center",
    userSelect: "none",
    whiteSpace: "nowrap",
    verticalAlign: "middle",
    outline: "none",
    ...btnStyles,
    ...isMobile ? !p ? { position: "fixed", top: 3, left: 3, zIndex: "popover" } : {} : { display: "none" },
    ...styles
  };
  const _icon = icon ? (0, import_utils18.runIfFn)(icon, {
    isOpen
  }) : /* @__PURE__ */ (0, import_jsx_runtime29.jsx)(HamburgerIcon, { "aria-hidden": "true", focusable: "false" });
  return /* @__PURE__ */ (0, import_jsx_runtime29.jsx)(
    import_react36.chakra.button,
    {
      __css: buttonStyles,
      padding: "0",
      borderRadius: isRound ? "full" : void 0,
      "aria-label": isOpen ? "Close sidebar" : "Open sidebar",
      "data-state": isOpen ? "open" : "closed",
      ...rest,
      ...getButtonProps(props),
      sx,
      className: (0, import_utils18.cx)("sui-sidebar__toggle-button", props.className),
      children: _icon
    }
  );
};
var SidebarOverlay = (props) => {
  const { onClose, isOpen, isMobile } = useSidebarContext();
  const styles = useSidebarStyles();
  return /* @__PURE__ */ (0, import_jsx_runtime29.jsx)(import_react36.Portal, { children: /* @__PURE__ */ (0, import_jsx_runtime29.jsx)(import_framer_motion7.AnimatePresence, { children: isMobile && isOpen && /* @__PURE__ */ (0, import_jsx_runtime29.jsx)(
    MotionBox2,
    {
      animate: isOpen ? "enter" : "exit",
      initial: "exit",
      variants: {
        enter: { opacity: 1 },
        exit: { opacity: 0 }
      },
      position: "fixed",
      top: "0",
      right: "0",
      bottom: "0",
      left: "0",
      zIndex: "overlay",
      ...props,
      onClick: onClose,
      __css: styles.overlay
    }
  ) }) });
};
SidebarToggleButton.displayName = "SidebarToggleButton";
var SidebarSection = (props) => {
  const { direction = "column", ...rest } = props;
  const styles = useSidebarStyles();
  const sectionStyles = {
    display: "flex",
    flexDirection: direction,
    ...styles.section
  };
  return /* @__PURE__ */ (0, import_jsx_runtime29.jsx)(
    import_react36.chakra.div,
    {
      __css: sectionStyles,
      ...rest,
      className: (0, import_utils18.cx)("sui-sidebar__section", props.className)
    }
  );
};
SidebarSection.displayName = "SidebarSection";

// src/sidebar/nav.tsx
var import_react37 = require("@chakra-ui/react");
var import_utils19 = require("@chakra-ui/utils");
var import_jsx_runtime30 = require("react/jsx-runtime");
var Nav = (props) => {
  const styles = (0, import_react37.useStyleConfig)("SuiNav", props);
  const { children, spacing = 0, orientation = "vertical", ...rest } = props;
  const containerStyles = {
    "& > *:not(style) ~ *:not(style)": {
      mt: orientation === "vertical" ? spacing : void 0,
      me: orientation === "horizontal" ? spacing : void 0
    },
    ...styles
  };
  return /* @__PURE__ */ (0, import_jsx_runtime30.jsx)(
    import_react37.chakra.nav,
    {
      ...rest,
      __css: containerStyles,
      className: (0, import_utils19.cx)("sui-nav", props.className),
      "data-orientation": orientation,
      children
    }
  );
};
Nav.displayName = "Nav";

// src/sidebar/nav-group.tsx
var React13 = __toESM(require("react"));
var import_react38 = require("@chakra-ui/react");
var import_utils20 = require("@chakra-ui/utils");

// src/sidebar/nav-context.tsx
var [NavGroupStylesProvider, useNavGroupStyles] = createContext7({
  name: `NavGroupStylesContext`,
  hookName: `useNavItemStyles`,
  providerName: "<NavItem />"
});
var [NavItemStylesProvider, useNavItemStyles] = createContext7({
  name: `NavItemStylesContext`,
  hookName: `useNavItemStyles`,
  providerName: "<NavItem />"
});

// src/sidebar/nav-group.tsx
var import_jsx_runtime31 = require("react/jsx-runtime");
var NavGroupIcon = (props) => {
  const { children, className, ...rest } = props;
  const _children = React13.isValidElement(children) ? React13.cloneElement(children, {
    "aria-hidden": true,
    focusable: false
  }) : children;
  const _className = (0, import_utils20.cx)("sui-nav-group__icon", className);
  return /* @__PURE__ */ (0, import_jsx_runtime31.jsx)(
    import_react38.chakra.span,
    {
      display: "inline-flex",
      alignSelf: "center",
      flexShrink: 0,
      ...rest,
      className: _className,
      children: _children
    }
  );
};
var NavGroupTitle = (props) => {
  const {
    leftIcon,
    collapseIcon = ({ isOpen: isOpen2 }) => isOpen2 ? /* @__PURE__ */ (0, import_jsx_runtime31.jsx)(ChevronDownIcon, {}) : /* @__PURE__ */ (0, import_jsx_runtime31.jsx)(ChevronRightIcon, {}),
    children,
    ...rest
  } = props;
  const styles = useNavGroupStyles();
  const { getToggleProps, isOpen, isCollapsible } = useCollapseContext();
  const iconStyles = { marginEnd: 2, ...styles.icon };
  const toggleProps = getToggleProps(rest);
  return /* @__PURE__ */ (0, import_jsx_runtime31.jsxs)(
    import_react38.chakra.div,
    {
      ...toggleProps,
      __css: styles.title,
      className: (0, import_utils20.cx)(
        "saas-nav-group__title",
        props.className,
        toggleProps.className
      ),
      children: [
        leftIcon && /* @__PURE__ */ (0, import_jsx_runtime31.jsx)(NavGroupIcon, { __css: iconStyles, children: leftIcon }),
        /* @__PURE__ */ (0, import_jsx_runtime31.jsx)(import_react38.chakra.span, { flex: "1", children: (0, import_utils20.runIfFn)(children, { isOpen }) }),
        isCollapsible && /* @__PURE__ */ (0, import_jsx_runtime31.jsx)(NavGroupIcon, { children: (0, import_utils20.runIfFn)(collapseIcon, { isOpen }) })
      ]
    }
  );
};
NavGroupTitle.displayName = "NavGroupTitle";
var NavGroupContent = (props) => {
  const styles = useNavGroupStyles();
  return /* @__PURE__ */ (0, import_jsx_runtime31.jsx)(
    import_react38.chakra.div,
    {
      ...props,
      __css: styles.content,
      className: (0, import_utils20.cx)("sui-nav-group__content", props.className)
    }
  );
};
var NavGroup = (props) => {
  const {
    title,
    icon,
    isCollapsible = false,
    defaultIsOpen = true,
    onOpen,
    onClose,
    children,
    ...rest
  } = props;
  const styles = (0, import_react38.useMultiStyleConfig)("SuiNavGroup", props);
  const collapse = useCollapse({
    ...props,
    defaultIsOpen
  });
  const { getCollapseProps } = collapse;
  let header = title;
  if (typeof title === "string") {
    header = /* @__PURE__ */ (0, import_jsx_runtime31.jsx)(NavGroupTitle, { leftIcon: icon, children: title });
  }
  let content = /* @__PURE__ */ (0, import_jsx_runtime31.jsx)(NavGroupContent, { children });
  if (isCollapsible) {
    content = /* @__PURE__ */ (0, import_jsx_runtime31.jsx)(import_react38.Collapse, { ...getCollapseProps(), children: content });
  }
  return /* @__PURE__ */ (0, import_jsx_runtime31.jsx)(NavGroupStylesProvider, { value: styles, children: /* @__PURE__ */ (0, import_jsx_runtime31.jsx)(CollapseProvider, { value: collapse, children: /* @__PURE__ */ (0, import_jsx_runtime31.jsxs)(
    import_react38.chakra.div,
    {
      __css: {
        userSelect: "none",
        ...styles.container
      },
      ...rest,
      className: (0, import_utils20.cx)("sui-nav-group", props.className),
      role: "group",
      children: [
        header,
        content
      ]
    }
  ) }) });
};
NavGroup.displayName = "NavGroup";

// src/sidebar/nav-item.tsx
var React14 = __toESM(require("react"));
var import_react39 = require("@chakra-ui/react");
var import_utils21 = require("@chakra-ui/utils");
var import_jsx_runtime32 = require("react/jsx-runtime");
var NavItemLabel = (0, import_react39.forwardRef)(
  ({ children, ...props }, ref) => {
    const styles = useNavItemStyles();
    return /* @__PURE__ */ (0, import_jsx_runtime32.jsx)(
      import_react39.chakra.span,
      {
        ref,
        __css: styles.label,
        ...props,
        className: (0, import_utils21.cx)("sui-nav-item__label", props.className),
        children
      }
    );
  }
);
NavItemLabel.displayName = "NavItemLabel";
var NavItemIcon = (props) => {
  const styles = useNavItemStyles();
  const { className, children, ...rest } = props;
  const child = React14.Children.only(children);
  const clone = React14.isValidElement(child) ? React14.cloneElement(child, {
    focusable: "false",
    "aria-hidden": true
  }) : null;
  return /* @__PURE__ */ (0, import_jsx_runtime32.jsx)(
    import_react39.chakra.span,
    {
      ...rest,
      className: (0, import_utils21.cx)("sui-nav-item__icon", props.className),
      __css: {
        flexShrink: 0,
        ...styles.icon
      },
      children: clone
    }
  );
};
NavItemIcon.displayName = "NavItemIcon";
var NavItem = (0, import_react39.forwardRef)((props, ref) => {
  const {
    href,
    icon,
    inset,
    className,
    tooltipProps,
    isActive,
    children,
    ...rest
  } = (0, import_react39.omitThemingProps)(props);
  const Link3 = useLink();
  const { onClose, variant: sidebarVariant } = useSidebarContext() || {};
  const isCompact = sidebarVariant === "compact";
  const styles = (0, import_react39.useMultiStyleConfig)("SuiNavItem", props);
  let label = children;
  let tooltipLabel = tooltipProps == null ? void 0 : tooltipProps.label;
  if (typeof label === "string") {
    if (!tooltipLabel && isCompact) {
      tooltipLabel = label;
    }
    label = /* @__PURE__ */ (0, import_jsx_runtime32.jsx)(NavItemLabel, { children: label });
  }
  let as;
  if (href) {
    as = Link3;
  }
  const link = /* @__PURE__ */ (0, import_jsx_runtime32.jsx)(
    import_react39.chakra.a,
    {
      as,
      "aria-current": isActive ? "page" : void 0,
      ...rest,
      ref,
      href,
      className: "sui-nav-item__link",
      "data-active": (0, import_utils21.dataAttr)(isActive),
      __css: styles.link,
      children: /* @__PURE__ */ (0, import_jsx_runtime32.jsxs)(
        import_react39.chakra.span,
        {
          __css: {
            ...styles.inner,
            pl: inset
          },
          className: "sui-nav-item__inner",
          children: [
            icon && /* @__PURE__ */ (0, import_jsx_runtime32.jsx)(NavItemIcon, { children: icon }),
            label
          ]
        }
      )
    }
  );
  return /* @__PURE__ */ (0, import_jsx_runtime32.jsx)(NavItemStylesProvider, { value: styles, children: /* @__PURE__ */ (0, import_jsx_runtime32.jsx)(
    import_react39.Tooltip,
    {
      label: tooltipLabel,
      placement: "right",
      openDelay: 400,
      ...tooltipProps,
      children: /* @__PURE__ */ (0, import_jsx_runtime32.jsx)(
        import_react39.chakra.div,
        {
          __css: styles.item,
          onClick: onClose,
          "data-compact": (0, import_utils21.dataAttr)(isCompact),
          className: (0, import_utils21.cx)("sui-nav-item", className),
          children: link
        }
      )
    }
  ) });
});
NavItem.displayName = "NavItem";

// src/search-input/search-input.tsx
var React15 = __toESM(require("react"));
var import_react40 = require("@chakra-ui/react");
var import_utils22 = require("@chakra-ui/utils");
var import_jsx_runtime33 = require("react/jsx-runtime");
var SearchInput = (0, import_react40.forwardRef)(
  (props, ref) => {
    const {
      placeholder = "Search",
      value: valueProp,
      defaultValue: defaultValueProp,
      size,
      variant,
      width,
      icon,
      resetIcon,
      rightElement,
      onChange: onChangeProp,
      onReset: onResetProp,
      onKeyDown: onKeyDownProp,
      ...inputProps
    } = props;
    const styles = (0, import_react40.useMultiStyleConfig)("SuiSearchInput", props);
    const inputRef = React15.useRef(null);
    const [value, setValue] = (0, import_react40.useControllableState)({
      value: valueProp,
      defaultValue: defaultValueProp
    });
    const onChange = React15.useCallback(
      (e) => {
        setValue(e.target.value);
      },
      [setValue]
    );
    const onKeyDown = React15.useCallback(
      (event) => {
        if (event.key === "Escape") {
          setValue("");
          onReset();
        }
      },
      [onResetProp, setValue]
    );
    const onReset = () => {
      var _a;
      setValue("");
      onResetProp == null ? void 0 : onResetProp();
      (_a = inputRef.current) == null ? void 0 : _a.focus();
    };
    const btnSize = size === "lg" ? "sm" : "xs";
    const showReset = value && !props.isDisabled;
    return /* @__PURE__ */ (0, import_jsx_runtime33.jsxs)(import_react40.InputGroup, { size, width, children: [
      /* @__PURE__ */ (0, import_jsx_runtime33.jsx)(import_react40.InputLeftElement, { children: icon || /* @__PURE__ */ (0, import_jsx_runtime33.jsx)(SearchIcon, {}) }),
      /* @__PURE__ */ (0, import_jsx_runtime33.jsx)(
        import_react40.Input,
        {
          type: "text",
          placeholder,
          variant,
          size,
          value,
          ref: (0, import_react40.useMergeRefs)(ref, inputRef),
          sx: styles.input,
          onChange: (0, import_utils22.callAllHandlers)(onChange, onChangeProp),
          onKeyDown: (0, import_utils22.callAllHandlers)(onKeyDown, onKeyDownProp),
          ...inputProps
        }
      ),
      /* @__PURE__ */ (0, import_jsx_runtime33.jsx)(import_react40.InputRightElement, { children: showReset ? /* @__PURE__ */ (0, import_jsx_runtime33.jsx)(
        import_react40.IconButton,
        {
          onClick: onReset,
          size: btnSize,
          variant: "ghost",
          "aria-label": "Reset search",
          icon: resetIcon || /* @__PURE__ */ (0, import_jsx_runtime33.jsx)(CloseIcon, {}),
          sx: styles.reset
        }
      ) : rightElement })
    ] });
  }
);
SearchInput.displayName = "SearchInput";

// src/snackbar/use-snackbar.tsx
var React16 = __toESM(require("react"));
var import_react41 = require("@chakra-ui/react");
var import_utils23 = require("@chakra-ui/utils");
var import_jsx_runtime34 = require("react/jsx-runtime");
var AlertSpinner = (props) => {
  const styles = (0, import_react41.useMultiStyleConfig)("Alert", props);
  return /* @__PURE__ */ (0, import_jsx_runtime34.jsx)(
    import_react41.chakra.span,
    {
      display: "inherit",
      alignItems: "center",
      ...props,
      __css: styles.icon,
      children: /* @__PURE__ */ (0, import_jsx_runtime34.jsx)(import_react41.Spinner, { size: "sm" })
    }
  );
};
var Snackbar = (props) => {
  var _a, _b, _c;
  const {
    status,
    variant,
    id,
    title,
    icon,
    isClosable,
    onClose,
    description,
    action
  } = props;
  const theme = (0, import_react41.useTheme)();
  const defaultVariant = ((_c = (_b = (_a = theme.components) == null ? void 0 : _a.Alert) == null ? void 0 : _b.variants) == null ? void 0 : _c.snackbar) ? "snackbar" : "solid";
  const alertTitleId = typeof id !== "undefined" ? `toast-${id}-title` : void 0;
  return /* @__PURE__ */ (0, import_jsx_runtime34.jsxs)(
    import_react41.Alert,
    {
      status,
      variant: variant || defaultVariant,
      id: id !== "undefined" ? `toast-${id}` : void 0,
      alignItems: "start",
      borderRadius: "md",
      boxShadow: "lg",
      textAlign: "start",
      width: "auto",
      "aria-labelledby": alertTitleId,
      children: [
        icon === null ? null : icon ? icon : /* @__PURE__ */ (0, import_jsx_runtime34.jsx)(import_react41.AlertIcon, {}),
        /* @__PURE__ */ (0, import_jsx_runtime34.jsxs)(import_react41.chakra.div, { flex: "1", maxWidth: "100%", children: [
          title && /* @__PURE__ */ (0, import_jsx_runtime34.jsx)(import_react41.AlertTitle, { id: alertTitleId, children: title }),
          description && /* @__PURE__ */ (0, import_jsx_runtime34.jsx)(import_react41.AlertDescription, { display: "block", children: description })
        ] }),
        (action || isClosable) && /* @__PURE__ */ (0, import_jsx_runtime34.jsxs)(import_react41.ButtonGroup, { size: "xs", variant: "ghost", ms: "2", children: [
          action,
          isClosable && /* @__PURE__ */ (0, import_jsx_runtime34.jsx)(import_react41.CloseButton, { size: "sm", onClick: onClose })
        ] })
      ]
    }
  );
};
function useSnackbar(defaultOptions) {
  const toast = (0, import_react41.useToast)(defaultOptions);
  const parseOptions = React16.useCallback(
    (options) => {
      if (typeof options === "string") {
        return {
          title: options
        };
      }
      return options;
    },
    []
  );
  return React16.useMemo(() => {
    const snackbar = (options) => {
      const opts = parseOptions(options);
      import_react41.useToast;
      return toast({
        render: (props) => /* @__PURE__ */ (0, import_jsx_runtime34.jsx)(Snackbar, { ...defaultOptions, ...opts, ...props }),
        ...opts
      });
    };
    snackbar.info = (options) => snackbar({
      status: "info",
      ...parseOptions(options)
    });
    snackbar.success = (options) => snackbar({
      status: "success",
      ...parseOptions(options)
    });
    snackbar.error = (options) => snackbar({
      status: "error",
      ...parseOptions(options)
    });
    snackbar.promise = async (promise, { loading, success, error: error2 }) => {
      let toastId;
      if (loading) {
        const options = parseOptions(loading);
        toastId = snackbar({
          status: "info",
          duration: null,
          icon: /* @__PURE__ */ (0, import_jsx_runtime34.jsx)(AlertSpinner, {}),
          ...options
        });
      }
      return promise.then((result) => {
        const options = {
          status: "success",
          duration: 5e3,
          ...parseOptions((0, import_utils23.runIfFn)(success, result))
        };
        if (toastId) {
          snackbar.update(toastId, options);
        } else {
          snackbar(options);
        }
        return result;
      }).catch((e) => {
        const options = {
          title: e.name,
          description: e.description,
          status: "error",
          duration: 5e3,
          ...parseOptions((0, import_utils23.runIfFn)(error2, e))
        };
        if (toastId) {
          snackbar.update(toastId, options);
        } else {
          snackbar(options);
        }
        if (typeof error2 !== "function") {
          throw e;
        }
      });
    };
    snackbar.update = (toastId, options) => {
      return toast.update(toastId, {
        render: (props) => /* @__PURE__ */ (0, import_jsx_runtime34.jsx)(Snackbar, { ...defaultOptions, ...options, ...props }),
        ...options
      });
    };
    snackbar.isActive = toast.isActive;
    snackbar.close = toast.close;
    snackbar.closeAll = toast.closeAll;
    return snackbar;
  }, [toast, defaultOptions]);
}

// src/snackbar/standalone.tsx
var import_react42 = require("@chakra-ui/react");
var import_jsx_runtime35 = require("react/jsx-runtime");
var createStandAloneSnackbar = (options) => {
  const { ToastContainer: SnackbarContainer, toast: snackbar } = (0, import_react42.createStandaloneToast)({
    ...options,
    defaultOptions: {
      render: (props) => /* @__PURE__ */ (0, import_jsx_runtime35.jsx)(Snackbar, { ...props }),
      ...options.defaultOptions
    }
  });
  return {
    SnackbarContainer,
    snackbar
  };
};

// src/stepper/steps.tsx
var React18 = __toESM(require("react"));
var import_react43 = require("@chakra-ui/react");
var import_react_utils10 = require("@saas-ui/react-utils");
var import_utils24 = require("@chakra-ui/utils");

// src/stepper/use-stepper.tsx
var React17 = __toESM(require("react"));
var import_react_utils9 = require("@chakra-ui/react-utils");
var [StepperProvider, useStepperContext] = (0, import_react_utils9.createContext)({
  name: "StepperContext",
  errorMessage: "useStepperContext: `context` is undefined. Seems you forgot to wrap stepper components in `<Stepper />`"
});
function useStepper(props) {
  const { step, onChange } = props;
  const [activeIndex, setIndex] = React17.useState(0);
  const stepsRef = React17.useRef([]);
  const [, onUpdate] = React17.useState(Date.now());
  const registerStep = React17.useCallback(
    (name) => {
      const newSteps = [...stepsRef.current];
      if (newSteps.indexOf(name) === -1) {
        newSteps.push(name);
      }
      stepsRef.current = newSteps;
      onUpdate(Date.now());
    },
    [stepsRef, onUpdate]
  );
  const unregisterStep = (name) => {
    stepsRef.current = stepsRef.current.slice(stepsRef.current.indexOf(name), 1);
  };
  const setStep = (name) => {
    const i = stepsRef.current.indexOf(name);
    if (i !== -1) {
      setIndex(i);
    }
  };
  const nextStep = () => {
    setIndex(activeIndex + 1);
  };
  const prevStep = () => {
    setIndex(activeIndex - 1);
  };
  React17.useEffect(() => {
    if (typeof step === "string") {
      setStep(step);
    } else if (typeof step === "number") {
      setIndex(step);
    } else if (activeIndex === -1) {
      setIndex(0);
    }
  }, [step]);
  React17.useEffect(() => {
    onChange == null ? void 0 : onChange(activeIndex);
  }, [activeIndex, onChange]);
  const context = {
    stepsRef,
    activeStep: stepsRef.current[activeIndex],
    activeIndex,
    isFirstStep: activeIndex === 0,
    isLastStep: activeIndex === stepsRef.current.length - 1,
    isCompleted: activeIndex >= stepsRef.current.length,
    setIndex,
    setStep,
    nextStep,
    prevStep,
    registerStep,
    unregisterStep
  };
  return context;
}
function useStep(props) {
  const { name, isActive, isCompleted } = props;
  const { registerStep, unregisterStep, activeStep } = useStepperContext();
  React17.useEffect(() => {
    if (!name) {
      return;
    }
    registerStep(name);
    return () => {
      unregisterStep(name);
    };
  }, []);
  return {
    isActive: name ? activeStep === name : isActive,
    isCompleted
  };
}
function useStepperPrevButton({ label = "Back" } = {}) {
  const { isFirstStep, prevStep } = useStepperContext();
  return {
    isDisabled: isFirstStep,
    onClick: prevStep,
    children: label
  };
}
function useStepperNextButton({
  label = "Next",
  submitLabel = "Submit"
} = {}) {
  const { isLastStep, isCompleted, nextStep } = useStepperContext();
  return {
    isDisabled: isCompleted,
    onClick: nextStep,
    children: isLastStep ? submitLabel : label
  };
}

// src/stepper/steps.tsx
var import_jsx_runtime36 = require("react/jsx-runtime");
var [StylesProvider8, useStyles8] = (0, import_react43.createStylesContext)("Stepper");
var Steps = (0, import_react43.forwardRef)((props, ref) => {
  var _a, _b, _c, _d;
  const {
    children,
    orientation = "horizontal",
    index,
    step,
    onChange,
    variant,
    colorScheme,
    size,
    stepperProps,
    ...rest
  } = props;
  const styles = (0, import_react43.useMultiStyleConfig)("Stepper", props);
  const context = useStepper({
    step: step != null ? step : index,
    onChange
  });
  const { activeIndex } = context;
  const isVertical = orientation === "vertical";
  const steps = (0, import_react_utils10.getChildrenOfType)(children, StepsItem);
  const itemStyles = {
    position: "relative",
    ...styles.item
  };
  const elements = steps.reduce((memo, step2, i, arr) => {
    const item = React18.cloneElement(step2, {
      key: i,
      ...step2.props,
      isActive: activeIndex === i,
      isCompleted: step2.props.isCompleted || activeIndex > i
    });
    if (isVertical) {
      memo.push(
        /* @__PURE__ */ (0, import_jsx_runtime36.jsxs)(
          import_react43.chakra.div,
          {
            className: "sui-steps__item",
            __css: itemStyles,
            children: [
              item,
              /* @__PURE__ */ (0, import_jsx_runtime36.jsx)(StepsContent, { isOpen: activeIndex === i, orientation, children: step2.props.children }),
              i < arr.length - 1 ? /* @__PURE__ */ (0, import_jsx_runtime36.jsx)(import_react43.StepSeparator, {}) : null
            ]
          },
          `content-${i}`
        )
      );
    } else {
      memo.push(item);
    }
    return memo;
  }, []);
  const completed = (0, import_react_utils10.getChildOfType)(children, StepsCompleted);
  const hasContent = (_b = (_a = steps[activeIndex]) == null ? void 0 : _a.props) == null ? void 0 : _b.children;
  const content = activeIndex >= steps.length ? completed : !isVertical && hasContent ? /* @__PURE__ */ (0, import_jsx_runtime36.jsx)(StepsContent, { orientation, children: (_d = (_c = steps[activeIndex]) == null ? void 0 : _c.props) == null ? void 0 : _d.children }) : null;
  return /* @__PURE__ */ (0, import_jsx_runtime36.jsx)(StylesProvider8, { value: styles, children: /* @__PURE__ */ (0, import_jsx_runtime36.jsx)(StepperProvider, { value: context, children: /* @__PURE__ */ (0, import_jsx_runtime36.jsxs)(
    import_react43.chakra.div,
    {
      ref,
      __css: styles.container,
      ...rest,
      className: (0, import_utils24.cx)("sui-steps", props.className),
      children: [
        /* @__PURE__ */ (0, import_jsx_runtime36.jsx)(
          import_react43.Stepper,
          {
            index: activeIndex,
            orientation,
            variant,
            colorScheme,
            size,
            ...stepperProps,
            children: elements
          }
        ),
        content
      ]
    }
  ) }) });
});
Steps.displayName = "Steps";
var StepsItem = (props) => {
  const { render, icon, title, description, ...rest } = props;
  const step = useStep(rest);
  if (render) {
    return render({
      ...step,
      ...props
    });
  }
  return /* @__PURE__ */ (0, import_jsx_runtime36.jsxs)(import_react43.Step, { children: [
    /* @__PURE__ */ (0, import_jsx_runtime36.jsx)(import_react43.StepIndicator, { children: /* @__PURE__ */ (0, import_jsx_runtime36.jsx)(
      import_react43.StepStatus,
      {
        complete: /* @__PURE__ */ (0, import_jsx_runtime36.jsx)(import_react43.StepIcon, {}),
        incomplete: /* @__PURE__ */ (0, import_jsx_runtime36.jsx)(import_react43.StepNumber, { children: icon }),
        active: /* @__PURE__ */ (0, import_jsx_runtime36.jsx)(import_react43.StepNumber, {})
      }
    ) }),
    /* @__PURE__ */ (0, import_jsx_runtime36.jsxs)(import_react43.Box, { flexShrink: "0", children: [
      /* @__PURE__ */ (0, import_jsx_runtime36.jsx)(import_react43.StepTitle, { children: title }),
      description && /* @__PURE__ */ (0, import_jsx_runtime36.jsx)(import_react43.StepDescription, { children: description })
    ] }),
    /* @__PURE__ */ (0, import_jsx_runtime36.jsx)(import_react43.StepSeparator, {})
  ] });
};
StepsItem.displayName = "StepsItem";
var StepsContent = (props) => {
  const { children, isOpen = true, orientation = "horizontal", ...rest } = props;
  const styles = useStyles8();
  return /* @__PURE__ */ (0, import_jsx_runtime36.jsx)(
    import_react43.chakra.div,
    {
      ...rest,
      __css: styles.content,
      className: (0, import_utils24.cx)("sui-steps__content", props.className),
      "data-orientation": orientation,
      children: orientation === "vertical" ? /* @__PURE__ */ (0, import_jsx_runtime36.jsx)(
        import_react43.Collapse,
        {
          in: isOpen,
          style: { overflow: isOpen ? "visible" : "hidden" },
          children: /* @__PURE__ */ (0, import_jsx_runtime36.jsx)(import_react43.chakra.div, { p: "2px", children: isOpen ? children : null })
        }
      ) : children
    }
  );
};
StepsContent.displayName = "StepsContent";
var StepsCompleted = (props) => {
  const completedStyles = {};
  return /* @__PURE__ */ (0, import_jsx_runtime36.jsx)(
    import_react43.chakra.div,
    {
      __css: completedStyles,
      ...props,
      className: (0, import_utils24.cx)("sui-steps__completed", props.className)
    }
  );
};
StepsCompleted.displayName = "StepsCompleted";

// src/web3-address/address.tsx
var import_react44 = require("@chakra-ui/react");
var import_utils25 = require("@chakra-ui/utils");
var import_jsx_runtime37 = require("react/jsx-runtime");
var Web3Address = (props) => {
  const { address, startLength = 6, endLength = 4, ...rest } = props;
  const start = address.slice(0, startLength);
  const end = address.slice(address.length - endLength, address.length);
  return /* @__PURE__ */ (0, import_jsx_runtime37.jsxs)(import_react44.chakra.span, { ...rest, className: (0, import_utils25.cx)("sui-web3-address", props.className), children: [
    start,
    "...",
    end
  ] });
};
Web3Address.displayName = "Web3Address";

// src/timeline/timeline.tsx
var import_react45 = require("@chakra-ui/react");
var import_utils26 = require("@chakra-ui/utils");
var import_jsx_runtime38 = require("react/jsx-runtime");
var [StylesProvider9, useStyles9] = (0, import_react45.createStylesContext)("SuiTimeline");
var useTimelineStyles = useStyles9;
var Timeline = (props) => {
  const { children, orientation = "vertical", ...rest } = props;
  const styles = (0, import_react45.useMultiStyleConfig)("SuiTimeline", props);
  const timelineStyles = {
    position: "relative",
    listStyle: "none",
    ...styles.container
  };
  return /* @__PURE__ */ (0, import_jsx_runtime38.jsx)(StylesProvider9, { value: styles, children: /* @__PURE__ */ (0, import_jsx_runtime38.jsx)(
    import_react45.chakra.ul,
    {
      ...rest,
      __css: timelineStyles,
      className: (0, import_utils26.cx)("sui-timeline", props.className),
      "data-orientation": "vertical",
      children
    }
  ) });
};
Timeline.displayName = "Timeline";
var TimelineItem = (0, import_react45.forwardRef)(
  (props, ref) => {
    const { children, ...rest } = props;
    const styles = useStyles9();
    return /* @__PURE__ */ (0, import_jsx_runtime38.jsx)(
      import_react45.chakra.li,
      {
        ...rest,
        ref,
        __css: styles.item,
        className: (0, import_utils26.cx)("sui-timeline__item", props.className),
        children
      }
    );
  }
);
TimelineItem.displayName = "TimelineItem";
var TimelineContent = (props) => {
  const { children, ...rest } = props;
  const styles = useStyles9();
  const contentStyles = {
    flex: 1,
    ...styles.content
  };
  return /* @__PURE__ */ (0, import_jsx_runtime38.jsx)(
    import_react45.chakra.div,
    {
      ...rest,
      __css: contentStyles,
      className: (0, import_utils26.cx)("sui-timeline__content", props.className),
      children
    }
  );
};
TimelineContent.displayName = "TimelineContent";
var TimelineSeparator = (props) => {
  const { children, ...rest } = props;
  const styles = useStyles9();
  const separatorStyles = {
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    flexShrink: 0,
    ...styles.separator
  };
  return /* @__PURE__ */ (0, import_jsx_runtime38.jsx)(
    import_react45.chakra.div,
    {
      ...rest,
      __css: separatorStyles,
      className: (0, import_utils26.cx)("sui-timeline__separator", props.className),
      children
    }
  );
};
TimelineSeparator.displayName = "TimelineSeparator";
var TimelineDot = (props) => {
  const { children, ...rest } = props;
  const styles = useStyles9();
  const dotStyles = {
    ...styles.dot
  };
  return /* @__PURE__ */ (0, import_jsx_runtime38.jsx)(
    import_react45.chakra.div,
    {
      ...rest,
      __css: dotStyles,
      className: (0, import_utils26.cx)("sui-timeline__dot", props.className),
      children
    }
  );
};
TimelineDot.displayName = "TimelineDot";
var TimelineIcon = (props) => {
  const { children, ...rest } = props;
  const styles = useStyles9();
  const iconStyles = {
    display: "flex",
    alignItems: "center",
    zIndex: 1,
    ...styles.icon
  };
  return /* @__PURE__ */ (0, import_jsx_runtime38.jsx)(
    import_react45.chakra.div,
    {
      ...rest,
      __css: iconStyles,
      className: (0, import_utils26.cx)("sui-timeline__icon", props.className),
      "data-dot": (0, import_utils26.dataAttr)(!children),
      children: children || /* @__PURE__ */ (0, import_jsx_runtime38.jsx)(TimelineDot, {})
    }
  );
};
TimelineIcon.displayName = "TimelineIcon";
var TimelineTrack = (props) => {
  const { children, ...rest } = props;
  const styles = useStyles9();
  return /* @__PURE__ */ (0, import_jsx_runtime38.jsx)(
    import_react45.chakra.div,
    {
      ...rest,
      __css: styles.track,
      className: (0, import_utils26.cx)("sui-timeline__track", props.className),
      children
    }
  );
};
TimelineTrack.displayName = "TimelineTrack";

// src/icon-badge/icon-badge.tsx
var import_react46 = require("@chakra-ui/react");
var import_utils27 = require("@chakra-ui/utils");
var import_react47 = require("react");
var import_jsx_runtime39 = require("react/jsx-runtime");
var IconBadge = (0, import_react46.forwardRef)((props, ref) => {
  const { icon, children, isRound, "aria-label": ariaLabel, ...rest } = props;
  const styles = (0, import_react46.useStyleConfig)("SuiIconBadge", props);
  const itemProps = (0, import_react46.omitThemingProps)(rest);
  const element = icon || children;
  const _children = (0, import_react47.isValidElement)(element) ? (0, import_react47.cloneElement)(element, {
    "aria-hidden": true,
    focusable: false
  }) : null;
  const __css = {
    display: "inline-flex",
    alignItems: "center",
    justifyContent: "center",
    ...styles
  };
  return /* @__PURE__ */ (0, import_jsx_runtime39.jsx)(
    import_react46.chakra.div,
    {
      ref,
      __css,
      borderRadius: isRound ? "full" : void 0,
      "aria-label": ariaLabel,
      ...itemProps,
      className: (0, import_utils27.cx)("sui-icon-badge", props.className),
      children: _children
    }
  );
});

// src/utils/split-props.ts
var splitProps = (props, keys) => {
  const picked = {};
  const omitted = { ...props };
  if (typeof keys === "function") {
    for (const key in props) {
      if (keys(key)) {
        picked[key] = props[key];
        delete omitted[key];
      }
    }
  } else {
    for (const key of keys) {
      picked[key] = props[key];
      delete omitted[key];
    }
  }
  return [picked, omitted];
};

// src/utils/index.ts
var import_utils28 = require("@chakra-ui/utils");
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AppShell,
  Banner,
  BannerActions,
  BannerCloseButton,
  BannerContent,
  BannerDescription,
  BannerIcon,
  BannerTitle,
  Br,
  CalendarIcon,
  CheckIcon,
  ChevronDownIcon,
  ChevronLeftIcon,
  ChevronRightIcon,
  ChevronUpIcon,
  CloseIcon,
  CollapseProvider,
  Command,
  ContextMenu,
  ContextMenuItem,
  ContextMenuList,
  ContextMenuProvider,
  ContextMenuTrigger,
  EmptyState,
  EmptyStateActions,
  EmptyStateBody,
  EmptyStateContainer,
  EmptyStateDescription,
  EmptyStateFooter,
  EmptyStateIcon,
  EmptyStateTitle,
  ErrorBoundary,
  FilterIcon,
  HamburgerIcon,
  IconBadge,
  Link,
  LoadingOverlay,
  LoadingSpinner,
  LoadingText,
  MinusIcon,
  Nav,
  NavGroup,
  NavGroupContent,
  NavGroupStylesProvider,
  NavGroupTitle,
  NavItem,
  NavItemLabel,
  NavItemStylesProvider,
  Navbar,
  NavbarBrand,
  NavbarContent,
  NavbarItem,
  NavbarLink,
  NavbarProvider,
  OverflowMenu,
  Persona,
  PersonaAvatar,
  PersonaContainer,
  PersonaDetails,
  PersonaLabel,
  PersonaSecondaryLabel,
  PersonaTertiaryLabel,
  PlusIcon,
  Presence,
  Property,
  PropertyLabel,
  PropertyList,
  PropertyValue,
  SaasContext,
  SaasProvider,
  SearchIcon,
  SearchInput,
  Sidebar,
  SidebarOverlay,
  SidebarProvider,
  SidebarSection,
  SidebarStylesProvider,
  SidebarToggleButton,
  Snackbar,
  StepperProvider,
  Steps,
  StepsCompleted,
  StepsContent,
  StepsItem,
  StructuredList,
  StructuredListButton,
  StructuredListCell,
  StructuredListHeader,
  StructuredListIcon,
  StructuredListItem,
  Timeline,
  TimelineContent,
  TimelineDot,
  TimelineIcon,
  TimelineItem,
  TimelineSeparator,
  TimelineTrack,
  ViewIcon,
  ViewOffIcon,
  Web3Address,
  createIcon,
  createStandAloneSnackbar,
  defaultPresenceTokens,
  splitProps,
  useCollapse,
  useCollapseContext,
  useContextMenu,
  useContextMenuContext,
  useEmptyStateStyles,
  useLink,
  useNavGroupStyles,
  useNavItemStyles,
  useNavbar,
  useNavbarContext,
  useSaas,
  useSidebarContext,
  useSidebarStyles,
  useSidebarToggleButton,
  useSnackbar,
  useStep,
  useStepper,
  useStepperContext,
  useStepperNextButton,
  useStepperPrevButton,
  useTimelineStyles
});
